// client/src/pages/admin/AdminDashboard.jsx
import { useEffect, useState } from 'react';
import { Link } from 'react-router-dom';
import api from '../../services/api';
import toast from 'react-hot-toast';
import { useAuth } from '../../context/AuthContext';

const AdminDashboard = () => {
  const { isDemoMode } = useAuth();
  const [stats, setStats] = useState({
    totalTasks: 0,
    totalUsers: 0,
    pendingWithdrawals: 0,
    totalSubmissions: 0,
    totalWorkers: 0,
    totalBuyers: 0,
    totalRevenue: 0,
  });
  const [loading, setLoading] = useState(true);
  const [lastUpdate, setLastUpdate] = useState(new Date());
  const [refreshing, setRefreshing] = useState(false);

  // Enhanced demo data with more realistic numbers
  const demoTasks = [
    { _id: '1', title: 'Data Entry - Product Catalog', status: 'open', submissionsCount: 12 },
    { _id: '2', title: 'Content Writing - Blog Posts', status: 'in_progress', submissionsCount: 8 },
    { _id: '3', title: 'Image Tagging for AI Training', status: 'completed', submissionsCount: 30 },
    { _id: '4', title: 'Social Media Content Creation', status: 'open', submissionsCount: 15 },
    { _id: '5', title: 'Translation - English to Spanish', status: 'completed', submissionsCount: 10 },
    { _id: '6', title: 'Product Reviews Writing', status: 'open', submissionsCount: 18 },
  ];

  const demoUsers = [
    { _id: '1', email: 'admin@microearn.com', role: 'admin', totalSpent: 0, coins: 10000 },
    { _id: '2', email: 'buyer@microearn.com', role: 'buyer', totalSpent: 4300, coins: 5000 },
    { _id: '3', email: 'worker@microearn.com', role: 'worker', totalSpent: 0, coins: 1000 },
    { _id: '4', email: 'demo@microearn.com', role: 'worker', totalSpent: 0, coins: 1000 },
    { _id: '5', email: 'user1@example.com', role: 'worker', totalSpent: 0, coins: 500 },
    { _id: '6', email: 'user2@example.com', role: 'buyer', totalSpent: 2000, coins: 3000 },
    { _id: '7', email: 'user3@example.com', role: 'worker', totalSpent: 0, coins: 750 },
    { _id: '8', email: 'user4@example.com', role: 'buyer', totalSpent: 1500, coins: 2500 },
  ];

  const demoWithdrawals = [
    { _id: '1', userId: '3', amount: 500, status: 'pending', createdAt: '2024-02-01' },
    { _id: '2', userId: '4', amount: 300, status: 'pending', createdAt: '2024-02-02' },
    { _id: '3', userId: '7', amount: 200, status: 'completed', createdAt: '2024-01-30' },
  ];

  useEffect(() => {
    if (isDemoMode) {
      // In demo mode, use demo data with dynamic updates
      const pending = demoWithdrawals.filter(w => w.status === 'pending').length;
      const workers = demoUsers.filter(u => u.role === 'worker').length;
      const buyers = demoUsers.filter(u => u.role === 'buyer').length;
      const totalSubmissions = demoTasks.reduce((sum, task) => sum + task.submissionsCount, 0);

      setStats({
        totalTasks: demoTasks.length,
        totalUsers: demoUsers.length,
        pendingWithdrawals: pending,
        totalSubmissions,
        totalWorkers: workers,
        totalBuyers: buyers,
        totalRevenue: demoUsers.reduce((acc, curr) => acc + (curr.totalSpent || 0), 0),
      });
      setLoading(false);
      setLastUpdate(new Date());
    } else {
      fetchStats();
    }
  }, [isDemoMode]);

  // Auto-refresh functionality
  useEffect(() => {
    const interval = setInterval(() => {
      if (isDemoMode) {
        // Simulate real-time updates in demo mode
        setStats(prev => ({
          ...prev,
          totalUsers: prev.totalUsers + Math.floor(Math.random() * 3), // Simulate new users
          totalSubmissions: prev.totalSubmissions + Math.floor(Math.random() * 5), // Simulate new submissions
        }));
        setLastUpdate(new Date());
      }
    }, 30000); // Update every 30 seconds

    return () => clearInterval(interval);
  }, [isDemoMode]);

  const fetchStats = async () => {
    try {
      setRefreshing(true);
      const [tasksRes, withdrawalsRes, usersRes] = await Promise.all([
        api.get('/tasks/available?limit=1000'),
        api.get('/withdrawals'),
        api.get('/auth/users'),
      ]);

      const tasks = tasksRes.data.tasks || [];
      const withdrawals = withdrawalsRes.data.withdrawals || [];
      const users = usersRes.data.users || [];

      const pending = withdrawals.filter(w => w.status === 'pending').length;
      const workers = users.filter(u => u.role === 'worker').length;
      const buyers = users.filter(u => u.role === 'buyer').length;

      setStats({
        totalTasks: tasks.length,
        totalUsers: users.length,
        pendingWithdrawals: pending,
        totalSubmissions: 320, // Placeholder - would come from API
        totalWorkers: workers,
        totalBuyers: buyers,
        totalRevenue: users.reduce((acc, curr) => acc + (curr.totalSpent || 0), 0),
      });
      setLastUpdate(new Date());
    } catch (error) {
      console.error('Failed to load stats:', error);
      toast.error('Failed to load dashboard stats');
      // Set placeholder data on error
      setStats({
        totalTasks: 6,
        totalUsers: 150,
        pendingWithdrawals: 3,
        totalSubmissions: 320,
        totalWorkers: 95,
        totalBuyers: 55,
        totalRevenue: 12500,
      });
    } finally {
      setLoading(false);
      setRefreshing(false);
    }
  };

  const handleRefresh = () => {
    if (isDemoMode) {
      // Simulate refresh with random changes
      setStats(prev => ({
        ...prev,
        totalTasks: prev.totalTasks + Math.floor(Math.random() * 2),
        pendingWithdrawals: Math.max(0, prev.pendingWithdrawals + Math.floor(Math.random() * 3) - 1),
      }));
      setLastUpdate(new Date());
      toast.success('Dashboard refreshed! (Demo Mode)');
    } else {
      fetchStats();
    }
  };

  const getGrowthPercentage = (current, previous) => {
    if (previous === 0) return 0;
    return Math.round(((current - previous) / previous) * 100);
  };

  const formatLastUpdate = (date) => {
    const now = new Date();
    const diff = now - date;
    const minutes = Math.floor(diff / 60000);
    const seconds = Math.floor((diff % 60000) / 1000);
    
    if (minutes < 1) {
      return `${seconds}s ago`;
    } else if (minutes < 60) {
      return `${minutes}m ${seconds}s ago`;
    } else {
      return `${Math.floor(minutes / 60)}h ${minutes % 60}m ago`;
    }
  };

  if (loading) {
    return (
      <div className="flex items-center justify-center h-64">
        <div className="animate-spin rounded-full h-12 w-12 border-b-2 border-blue-600"></div>
      </div>
    );
  }

  return (
    <div className="max-w-7xl mx-auto space-y-8 animate-in fade-in slide-in-from-bottom-6 duration-700">
      {/* Header with Refresh */}
      <div className="flex flex-col md:flex-row justify-between items-start md:items-center gap-4">
        <div>
          <h1 className="text-4xl font-black text-gray-900 uppercase tracking-tighter">Admin Dashboard</h1>
          <p className="text-gray-500 font-medium mt-2">Real-time platform overview and management</p>
        </div>
        <div className="flex items-center space-x-4">
          <button
            onClick={handleRefresh}
            disabled={refreshing}
            className="flex items-center space-x-2 px-4 py-2 bg-white border border-gray-200 rounded-xl hover:bg-gray-50 transition-all duration-300 disabled:opacity-50"
          >
            <svg className={`w-4 h-4 ${refreshing ? 'animate-spin' : ''}`} fill="none" stroke="currentColor" viewBox="0 0 24 24">
              <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M4 4v5h.582m15.356 2A8.001 8.001 0 004.582 15.356H16m-4-4v5m0 0h6m-6 4h6" />
            </svg>
            <span className="text-sm font-medium">Refresh</span>
          </button>
          <div className="flex items-center space-x-2 text-xs text-gray-400">
            <svg className="w-3 h-3" fill="none" stroke="currentColor" viewBox="0 0 24 24">
              <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z" />
            </svg>
            <span>Last updated: {formatLastUpdate(lastUpdate)}</span>
          </div>
        </div>
      </div>

      {/* Main Stats Grid with Animations */}
      <div className="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-4 gap-6">
        {/* Total Users */}
        <div className="bg-gradient-to-br from-blue-50 to-cyan-50 border border-blue-200 rounded-2xl p-6 hover:shadow-lg transition-all duration-300 transform hover:scale-105 relative overflow-hidden">
          <div className="absolute top-0 right-0 w-20 h-20 bg-gradient-to-br from-blue-400/20 to-cyan-400/20 rounded-full -mr-10 -mt-10">
            <div className="w-full h-full flex items-center justify-center">
              <svg className="w-8 h-8 text-blue-600" fill="none" stroke="currentColor" viewBox="0 0 24 24">
                <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M12 4.354a4 4 0 110 5.292M15 21H3v-1a6 6 0 0112 0v1zm0 0h6v-1a6 6 0 00-9-5.197M13 7a4 4 0 11-8 0 4 4 0 018 0z" />
              </svg>
            </div>
          </div>
          <div className="flex items-center justify-between mb-4">
            <div className="p-3 bg-blue-500 rounded-xl shadow-lg">
              <svg className="w-6 h-6 text-white" fill="none" stroke="currentColor" viewBox="0 0 24 24">
                <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M12 4.354a4 4 0 110 5.292M15 21H3v-1a6 6 0 0112 0v1zm0 0h6v-1a6 6 0 00-9-5.197M13 7a4 4 0 11-8 0 4 4 0 018 0z" />
              </svg>
            </div>
            <span className="text-xs font-semibold text-blue-600 bg-blue-100 px-2 py-1 rounded-full animate-pulse">
              +{getGrowthPercentage(stats.totalUsers, 150)}%
            </span>
          </div>
          <div className="text-gray-700 text-sm font-medium mb-1">Total Users</div>
          <div className="text-3xl font-black text-blue-800 animate-in fade-in duration-1000">{stats.totalUsers}</div>
          <div className="mt-2 text-xs text-gray-600 font-medium">{stats.totalWorkers} Workers • {stats.totalBuyers} Buyers</div>
        </div>

        {/* Total Tasks */}
        <div className="bg-gradient-to-br from-purple-50 to-pink-50 border border-purple-200 rounded-2xl p-6 hover:shadow-lg transition-all duration-300 transform hover:scale-105 relative overflow-hidden">
          <div className="absolute top-0 right-0 w-20 h-20 bg-gradient-to-br from-purple-400/20 to-pink-400/20 rounded-full -mr-10 -mt-10">
            <div className="w-full h-full flex items-center justify-center">
              <svg className="w-8 h-8 text-purple-600" fill="none" stroke="currentColor" viewBox="0 0 24 24">
                <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M9 5H7a2 2 0 00-2 2v12a2 2 0 002 2h10a2 2 0 002-2V7a2 2 0 00-2-2h-2M9 5a2 2 0 002 2h2a2 2 0 002-2M9 5a2 2 0 012-2h2a2 2 0 012 2" />
              </svg>
            </div>
          </div>
          <div className="flex items-center justify-between mb-4">
            <div className="p-3 bg-purple-500 rounded-xl shadow-lg">
              <svg className="w-6 h-6 text-white" fill="none" stroke="currentColor" viewBox="0 0 24 24">
                <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M9 5H7a2 2 0 00-2 2v12a2 2 0 002 2h10a2 2 0 002-2V7a2 2 0 00-2-2h-2M9 5a2 2 0 002 2h2a2 2 0 002-2M9 5a2 2 0 012-2h2a2 2 0 012 2" />
              </svg>
            </div>
            <span className="text-xs font-semibold text-purple-600 bg-purple-100 px-2 py-1 rounded-full animate-pulse">
              Active
            </span>
          </div>
          <Link to="/admin/tasks" className="text-xs font-semibold text-purple-600 hover:text-purple-700 transition-colors">
            View All →
          </Link>
          <div className="text-gray-700 text-sm font-medium mb-1">Total Tasks</div>
          <div className="text-3xl font-black text-purple-800 animate-in fade-in duration-1000">{stats.totalTasks}</div>
        </div>

        {/* Pending Withdrawals */}
        <div className="bg-gradient-to-br from-yellow-50 to-orange-50 border border-yellow-200 rounded-2xl p-6 hover:shadow-lg transition-all duration-300 transform hover:scale-105 relative overflow-hidden">
          <div className="absolute top-0 right-0 w-20 h-20 bg-gradient-to-br from-yellow-400/20 to-orange-400/20 rounded-full -mr-10 -mt-10">
            <div className="w-full h-full flex items-center justify-center">
              <svg className="w-8 h-8 text-yellow-600" fill="none" stroke="currentColor" viewBox="0 0 24 24">
                <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z" />
              </svg>
            </div>
          </div>
          <div className="flex items-center justify-between mb-4">
            <div className="p-3 bg-yellow-500 rounded-xl shadow-lg">
              <svg className="w-6 h-6 text-white" fill="none" stroke="currentColor" viewBox="0 0 24 24">
                <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z" />
              </svg>
            </div>
            <span className="text-xs font-semibold text-yellow-600 bg-yellow-100 px-2 py-1 rounded-full animate-pulse">
              {stats.pendingWithdrawals > 0 ? 'Action Required' : 'All Clear'}
            </span>
          </div>
          <Link to="/admin/withdrawals" className="text-xs font-semibold text-yellow-600 hover:text-yellow-700 transition-colors">
            Review →
          </Link>
          <div className="text-gray-700 text-sm font-medium mb-1">Pending Withdrawals</div>
          <div className="text-3xl font-black text-yellow-800 animate-in fade-in duration-1000">{stats.pendingWithdrawals}</div>
          <div className="mt-2 text-xs text-yellow-700 font-semibold">Requires attention</div>
        </div>

        {/* Total Submissions */}
        <div className="bg-gradient-to-br from-green-50 to-emerald-50 border border-green-200 rounded-2xl p-6 hover:shadow-lg transition-all duration-300 transform hover:scale-105 relative overflow-hidden">
          <div className="absolute top-0 right-0 w-20 h-20 bg-gradient-to-br from-green-400/20 to-emerald-400/20 rounded-full -mr-10 -mt-10">
            <div className="w-full h-full flex items-center justify-center">
              <svg className="w-8 h-8 text-green-600" fill="none" stroke="currentColor" viewBox="0 0 24 24">
                <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M9 12h6m-6 4h6m2 5H7a2 2 0 01-2-2V5a2 2 0 012-2h5.586a1 1 0 01.707.293l5.414 5.414a1 1 0 01.293.707V19a2 2 0 01-2 2z" />
              </svg>
            </div>
          </div>
          <div className="flex items-center justify-between mb-4">
            <div className="p-3 bg-green-500 rounded-xl shadow-lg">
              <svg className="w-6 h-6 text-white" fill="none" stroke="currentColor" viewBox="0 0 24 24">
                <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M9 12h6m-6 4h6m2 5H7a2 2 0 01-2-2V5a2 2 0 012-2h5.586a1 1 0 01.707.293l5.414 5.414a1 1 0 01.293.707V19a2 2 0 01-2 2z" />
              </svg>
            </div>
            <span className="text-xs font-semibold text-green-600 bg-green-100 px-2 py-1 rounded-full animate-pulse">
              Processing
            </span>
          </div>
          <Link to="/admin/submissions" className="text-xs font-semibold text-green-600 hover:text-green-700 transition-colors">
            Review →
          </Link>
          <div className="text-gray-700 text-sm font-medium mb-1">Total Submissions</div>
          <div className="text-3xl font-black text-green-800 animate-in fade-in duration-1000">{stats.totalSubmissions}</div>
        </div>
      </div>

      {/* Quick Actions */}
      <div className="grid grid-cols-1 md:grid-cols-4 gap-6">
        <Link to="/admin/withdrawals" className="group bg-white border-2 border-gray-200 hover:border-yellow-500 rounded-2xl p-6 transition-all duration-300 hover:shadow-lg">
          <div className="flex flex-col items-center text-center space-y-3">
            <div className="p-4 bg-yellow-100 group-hover:bg-yellow-500 rounded-xl transition-colors">
              <svg className="w-8 h-8 text-yellow-600 group-hover:text-white" fill="none" stroke="currentColor" viewBox="0 0 24 24">
                <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M17 9V7a2 2 0 00-2-2H5a2 2 0 00-2 2v6a2 2 0 002 2h2m2 4h10a2 2 0 002-2v-6a2 2 0 00-2-2H9a2 2 0 00-2-2h-2m7-5a2 2 0 11-4 0 2 2 0 014 0z" />
              </svg>
            </div>
            <div>
              <div className="font-bold text-gray-900">Withdrawals</div>
              <div className="text-sm text-gray-500">{stats.pendingWithdrawals} pending</div>
            </div>
          </div>
        </Link>

        <Link to="/admin/tasks" className="group bg-white border-2 border-gray-200 hover:border-purple-500 rounded-2xl p-6 transition-all duration-300 hover:shadow-lg">
          <div className="flex flex-col items-center text-center space-y-3">
            <div className="p-4 bg-purple-100 group-hover:bg-purple-500 rounded-xl transition-colors">
              <svg className="w-8 h-8 text-purple-600 group-hover:text-white" fill="none" stroke="currentColor" viewBox="0 0 24 24">
                <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M9 5H7a2 2 0 00-2 2v12a2 2 0 002 2h10a2 2 0 002-2V7a2 2 0 00-2-2h-2M9 5a2 2 0 002 2h2a2 2 0 002-2M9 5a2 2 0 012-2h2a2 2 0 012 2" />
              </svg>
            </div>
            <div>
              <div className="font-bold text-gray-900">All Tasks</div>
              <div className="text-sm text-gray-500">{stats.totalTasks} total</div>
            </div>
          </div>
        </Link>

        <Link to="/admin/submissions" className="group bg-white border-2 border-gray-200 hover:border-green-500 rounded-2xl p-6 transition-all duration-300 hover:shadow-lg">
          <div className="flex flex-col items-center text-center space-y-3">
            <div className="p-4 bg-green-100 group-hover:bg-green-500 rounded-xl transition-colors">
              <svg className="w-8 h-8 text-green-600 group-hover:text-white" fill="none" stroke="currentColor" viewBox="0 0 24 24">
                <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M9 12h6m-6 4h6m2 5H7a2 2 0 01-2-2V5a2 2 0 012-2h5.586a1 1 0 01.707.293l5.414 5.414a1 1 0 01.293.707V19a2 2 0 01-2 2z" />
              </svg>
            </div>
            <div>
              <div className="font-bold text-gray-900">Submissions</div>
              <div className="text-sm text-gray-500">{stats.totalSubmissions} total</div>
            </div>
          </div>
        </Link>

        <Link to="/admin/users" className="group bg-white border-2 border-gray-200 hover:border-blue-500 rounded-2xl p-6 transition-all duration-300 hover:shadow-lg">
          <div className="flex flex-col items-center text-center space-y-3">
            <div className="p-4 bg-blue-100 group-hover:bg-blue-500 rounded-xl transition-colors">
              <svg className="w-8 h-8 text-blue-600 group-hover:text-white" fill="none" stroke="currentColor" viewBox="0 0 24 24">
                <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M12 4.354a4 4 0 110 5.292M15 21H3v-1a6 6 0 0112 0v1zm0 0h6v-1a6 6 0 00-9-5.197M13 7a4 4 0 11-8 0 4 4 0 018 0z" />
              </svg>
            </div>
            <div>
              <div className="font-bold text-gray-900">Users</div>
              <div className="text-sm text-gray-500">{stats.totalUsers} total</div>
            </div>
          </div>
        </Link>
      </div>

      {/* Platform Analytics */}
      <div className="grid grid-cols-1 lg:grid-cols-2 gap-6">
        {/* Revenue Overview */}
        <div className="bg-white rounded-2xl p-6 shadow-sm border border-gray-200">
          <h3 className="text-xl font-bold mb-6">Revenue Overview</h3>
          <div className="space-y-4">
            <div className="flex items-center justify-between p-4 bg-gradient-to-r from-green-50 to-emerald-50 rounded-xl">
              <div>
                <div className="text-sm text-gray-600">Total Revenue</div>
                <div className="text-2xl font-black text-green-600">${stats.totalRevenue.toLocaleString()}</div>
              </div>
              <div className="p-3 bg-green-500 rounded-xl">
                <svg className="w-6 h-6 text-white" fill="none" stroke="currentColor" viewBox="0 0 24 24">
                  <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M13 7h8m0 0v8m0-8l-8 8-4-4-6 6" />
                </svg>
              </div>
            </div>
            <div className="grid grid-cols-2 gap-4">
              <div className="p-4 bg-blue-50 rounded-xl text-center">
                <div className="text-2xl font-black text-blue-600">{stats.totalWorkers}</div>
                <div className="text-xs text-gray-600 mt-1">Active Workers</div>
              </div>
              <div className="p-4 bg-purple-50 rounded-xl text-center">
                <div className="text-2xl font-black text-purple-600">{stats.totalBuyers}</div>
                <div className="text-xs text-gray-600 mt-1">Active Buyers</div>
              </div>
            </div>
          </div>
        </div>

        {/* Recent Activity */}
        <div className="bg-white rounded-2xl p-6 shadow-sm border border-gray-200">
          <h3 className="text-xl font-bold mb-6">Platform Health</h3>
          <div className="space-y-4">
            <div className="flex items-center justify-between p-4 bg-green-50 rounded-xl">
              <div className="flex items-center space-x-3">
                <div className="w-2 h-2 bg-green-500 rounded-full animate-pulse"></div>
                <span className="text-sm font-semibold text-gray-700">System Status</span>
              </div>
              <span className="text-sm font-bold text-green-600">Operational</span>
            </div>
            <div className="flex items-center justify-between p-4 bg-blue-50 rounded-xl">
              <div className="flex items-center space-x-3">
                <div className="w-2 h-2 bg-blue-500 rounded-full"></div>
                <span className="text-sm font-semibold text-gray-700">Active Tasks</span>
              </div>
              <span className="text-sm font-bold text-blue-600">{stats.totalTasks}</span>
            </div>
            <div className="flex items-center justify-between p-4 bg-yellow-50 rounded-xl">
              <div className="flex items-center space-x-3">
                <div className="w-2 h-2 bg-yellow-500 rounded-full"></div>
                <span className="text-sm font-semibold text-gray-700">Pending Actions</span>
              </div>
              <span className="text-sm font-bold text-yellow-600">{stats.pendingWithdrawals}</span>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default AdminDashboard;
