// client/src/pages/admin/AllUsers.jsx
import { useEffect, useState } from 'react';
import api from '../../services/api';
import toast from 'react-hot-toast';
import { useAuth } from '../../context/AuthContext';

const AllUsers = () => {
    const { isDemoMode } = useAuth();
    const [users, setUsers] = useState([]);
    const [loading, setLoading] = useState(true);
    const [filter, setFilter] = useState('all');
    const [searchQuery, setSearchQuery] = useState('');

    // Demo users data
    const demoUsers = [
        { 
            _id: '1', 
            displayName: 'Admin User', 
            email: 'admin@microearn.com', 
            role: 'admin', 
            totalSpent: 0, 
            coins: 10000,
            isActive: true,
            createdAt: '2024-01-01T00:00:00Z'
        },
        { 
            _id: '2', 
            displayName: 'Buyer User', 
            email: 'buyer@microearn.com', 
            role: 'buyer', 
            totalSpent: 4300, 
            coins: 5000,
            isActive: true,
            createdAt: '2024-01-02T00:00:00Z'
        },
        { 
            _id: '3', 
            displayName: 'Worker User', 
            email: 'worker@microearn.com', 
            role: 'worker', 
            totalSpent: 0, 
            coins: 1000,
            isActive: true,
            createdAt: '2024-01-03T00:00:00Z'
        },
        { 
            _id: '4', 
            displayName: 'Demo Worker', 
            email: 'demo@microearn.com', 
            role: 'worker', 
            totalSpent: 0, 
            coins: 1000,
            isActive: true,
            createdAt: '2024-01-04T00:00:00Z'
        },
        { 
            _id: '5', 
            displayName: 'User One', 
            email: 'user1@example.com', 
            role: 'worker', 
            totalSpent: 0, 
            coins: 500,
            isActive: true,
            createdAt: '2024-01-05T00:00:00Z'
        },
        { 
            _id: '6', 
            displayName: 'User Two', 
            email: 'user2@example.com', 
            role: 'buyer', 
            totalSpent: 2000, 
            coins: 3000,
            isActive: true,
            createdAt: '2024-01-06T00:00:00Z'
        },
        { 
            _id: '7', 
            displayName: 'User Three', 
            email: 'user3@example.com', 
            role: 'worker', 
            totalSpent: 0, 
            coins: 750,
            isActive: false,
            createdAt: '2024-01-07T00:00:00Z'
        },
        { 
            _id: '8', 
            displayName: 'User Four', 
            email: 'user4@example.com', 
            role: 'buyer', 
            totalSpent: 1500, 
            coins: 2500,
            isActive: true,
            createdAt: '2024-01-08T00:00:00Z'
        },
    ];

    useEffect(() => {
        if (isDemoMode) {
            // In demo mode, use demo data
            setUsers(demoUsers);
            setLoading(false);
        } else {
            fetchUsers();
        }
    }, [isDemoMode]);

    const fetchUsers = async () => {
        try {
            setLoading(true);
            const response = await api.get('/auth/users');
            setUsers(response.data.users || []);
        } catch (error) {
            toast.error('Failed to load users');
        } finally {
            setLoading(false);
        }
    };

    const handleStatusUpdate = async (userId, currentStatus) => {
        try {
            if (isDemoMode) {
                // Demo mode simulation
                setUsers(users.map(u => u._id === userId ? { ...u, isActive: !currentStatus } : u));
                toast.success(`User ${!currentStatus ? 'activated' : 'deactivated'} successfully! (Demo Mode)`);
            } else {
                const response = await api.patch('/auth/users/status', {
                    userId,
                    isActive: !currentStatus,
                });
                toast.success(response.data.message);
                // Update local state
                setUsers(users.map(u => u._id === userId ? { ...u, isActive: !currentStatus } : u));
            }
        } catch (error) {
            toast.error(error.response?.data?.message || 'Failed to update user status');
        }
    };

    const handleRoleUpdate = async (userId, newRole) => {
        try {
            if (isDemoMode) {
                // Demo mode simulation
                setUsers(users.map(u => u._id === userId ? { ...u, role: newRole } : u));
                toast.success(`User role updated to ${newRole}! (Demo Mode)`);
            } else {
                const response = await api.patch('/auth/users/role', {
                    userId,
                    role: newRole,
                });
                toast.success(response.data.message);
                // Update local state
                setUsers(users.map(u => u._id === userId ? { ...u, role: newRole } : u));
            }
        } catch (error) {
            toast.error(error.response?.data?.message || 'Failed to update user role');
        }
    };

    const filteredUsers = users.filter(user => {
        const matchesFilter = filter === 'all' || user.role === filter;
        const matchesSearch = user.displayName?.toLowerCase().includes(searchQuery.toLowerCase()) ||
            user.email?.toLowerCase().includes(searchQuery.toLowerCase());
        return matchesFilter && matchesSearch;
    });

    if (loading) {
        return (
            <div className="flex items-center justify-center h-64">
                <div className="animate-spin rounded-full h-12 w-12 border-b-2 border-blue-600"></div>
            </div>
        );
    }

    return (
        <div className="max-w-7xl mx-auto space-y-8 animate-in fade-in slide-in-from-bottom-6 duration-700">
            <div className="flex flex-col md:flex-row justify-between items-start md:items-center gap-6">
                <div>
                    <h2 className="text-4xl font-black text-gray-900 uppercase tracking-tighter">User Management</h2>
                    <p className="text-gray-500 font-medium mt-2">Manage platform users, roles, and permissions</p>
                </div>
                <div className="flex flex-col sm:flex-row gap-4 w-full md:w-auto">
                    <div className="relative">
                        <input
                            type="text"
                            placeholder="Search users..."
                            className="w-full sm:w-64 pl-12 pr-4 py-3 rounded-2xl bg-gray-50 border-2 border-transparent focus:bg-white focus:border-indigo-600 focus:ring-4 focus:ring-indigo-600/5 transition-all outline-none font-medium"
                            value={searchQuery}
                            onChange={(e) => setSearchQuery(e.target.value)}
                        />
                        <svg className="absolute left-4 top-1/2 -translate-y-1/2 w-5 h-5 text-gray-400" fill="none" stroke="currentColor" viewBox="0 0 24 24">
                            <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z" />
                        </svg>
                    </div>
                    <select
                        className="px-6 py-3 rounded-2xl bg-gray-50 border-2 border-transparent focus:bg-white focus:border-indigo-600 transition-all outline-none font-bold appearance-none cursor-pointer"
                        value={filter}
                        onChange={(e) => setFilter(e.target.value)}
                    >
                        <option value="all">All Users</option>
                        <option value="worker">Workers</option>
                        <option value="buyer">Buyers</option>
                        <option value="admin">Admins</option>
                    </select>
                </div>
            </div>

            <div className="bg-white rounded-3xl shadow-xl border border-gray-100 overflow-hidden">
                {loading ? (
                    <div className="flex items-center justify-center h-96">
                        <div className="animate-spin rounded-full h-12 w-12 border-b-2 border-indigo-600"></div>
                    </div>
                ) : filteredUsers.length === 0 ? (
                    <div className="flex flex-col items-center justify-center h-96 px-10 text-center">
                        <div className="w-20 h-20 bg-gray-50 text-gray-200 rounded-full flex items-center justify-center mb-6">
                            <svg className="w-10 h-10" fill="none" stroke="currentColor" viewBox="0 0 24 24">
                                <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M12 4.354a4 4 0 110 5.292M15 21H3v-1a6 6 0 0112 0v1zm0 0h6v-1a6 6 0 00-9-5.197M13 7a4 4 0 11-8 0 4 4 0 018 0z" />
                            </svg>
                        </div>
                        <h3 className="text-xl font-black text-gray-900 uppercase tracking-tighter mb-2">No Users Found</h3>
                        <p className="text-gray-400 font-medium">
                            {searchQuery ? 'Try adjusting your search terms' : 'No users match the current filter'}
                        </p>
                    </div>
                ) : (
                    <div className="overflow-x-auto">
                        <table className="min-w-full divide-y divide-gray-100">
                            <thead className="bg-gray-50">
                                <tr>
                                    <th className="px-8 py-4 text-left text-xs font-black text-gray-600 uppercase tracking-wider">User</th>
                                    <th className="px-6 py-4 text-left text-xs font-black text-gray-600 uppercase tracking-wider">Role</th>
                                    <th className="px-6 py-4 text-left text-xs font-black text-gray-600 uppercase tracking-wider">Coins</th>
                                    <th className="px-6 py-4 text-left text-xs font-black text-gray-600 uppercase tracking-wider">Status</th>
                                    <th className="px-8 py-4 text-left text-xs font-black text-gray-600 uppercase tracking-wider">Actions</th>
                                </tr>
                            </thead>
                            <tbody className="bg-white divide-y divide-gray-100">
                                {filteredUsers.map((user) => (
                                    <tr key={user._id} className="hover:bg-gray-50 transition-colors">
                                        <td className="px-8 py-6 whitespace-nowrap">
                                            <div className="flex items-center">
                                                <div className="flex-shrink-0 h-12 w-12">
                                                    <div className="h-12 w-12 rounded-full bg-gradient-to-br from-indigo-500 to-purple-600 flex items-center justify-center text-white font-bold text-lg">
                                                        {user.displayName?.charAt(0)?.toUpperCase() || 'U'}
                                                    </div>
                                                </div>
                                                <div className="ml-4">
                                                    <div className="text-sm font-bold text-gray-900">{user.displayName}</div>
                                                    <div className="text-sm text-gray-500">{user.email}</div>
                                                </div>
                                            </div>
                                        </td>
                                        <td className="px-6 py-6 whitespace-nowrap">
                                            <select
                                                value={user.role}
                                                onChange={(e) => handleRoleUpdate(user._id, e.target.value)}
                                                className={`text-sm font-bold px-3 py-1 rounded-full border-2 cursor-pointer transition-colors ${
                                                    user.role === 'admin' ? 'bg-red-50 text-red-700 border-red-200' :
                                                    user.role === 'buyer' ? 'bg-blue-50 text-blue-700 border-blue-200' :
                                                    'bg-green-50 text-green-700 border-green-200'
                                                }`}
                                                disabled={user.role === 'admin' && !isDemoMode}
                                            >
                                                <option value="worker">Worker</option>
                                                <option value="buyer">Buyer</option>
                                                <option value="admin">Admin</option>
                                            </select>
                                        </td>
                                        <td className="px-6 py-6 whitespace-nowrap">
                                            <div className="flex items-center space-x-2">
                                                <span className="text-lg font-black text-gray-900">{user.coins}</span>
                                                <span className="text-xs font-bold text-indigo-600 uppercase tracking-wider">Coins</span>
                                            </div>
                                        </td>
                                        <td className="px-6 py-6 whitespace-nowrap">
                                            <span className={`inline-flex items-center px-3 py-1 rounded-full text-xs font-bold uppercase tracking-wider ring-1 ${
                                                user.isActive 
                                                    ? 'bg-green-50 text-green-700 ring-green-200' 
                                                    : 'bg-red-50 text-red-700 ring-red-200'
                                            }`}>
                                                {user.isActive ? (
                                                    <>
                                                        <svg className="w-3 h-3 mr-1" fill="currentColor" viewBox="0 0 20 20">
                                                            <circle cx="10" cy="10" r="6" />
                                                        </svg>
                                                        Active
                                                    </>
                                                ) : (
                                                    <>
                                                        <svg className="w-3 h-3 mr-1" fill="currentColor" viewBox="0 0 20 20">
                                                            <circle cx="10" cy="10" r="6" />
                                                        </svg>
                                                        Blocked
                                                    </>
                                                )}
                                            </span>
                                        </td>
                                        <td className="px-8 py-6 whitespace-nowrap text-right">
                                            {user.role !== 'admin' && (
                                                <button
                                                    onClick={() => handleStatusUpdate(user._id, user.isActive)}
                                                    className={`px-4 py-2 rounded-xl text-sm font-bold uppercase tracking-wider transition-all transform hover:scale-105 active:scale-95 ${
                                                        user.isActive 
                                                            ? 'bg-red-50 text-red-600 hover:bg-red-100 ring-2 ring-red-200' 
                                                            : 'bg-green-50 text-green-600 hover:bg-green-100 ring-2 ring-green-200'
                                                    }`}
                                                >
                                                    {user.isActive ? (
                                                        <>
                                                            <svg className="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24">
                                                                <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M18.364 18.364A9 9 0 005.636 5.636m12.728 12.728A9 9 0 0018.364 5.636m-9 9h.01M6 6h12" />
                                                            </svg>
                                                            Block
                                                        </>
                                                    ) : (
                                                        <>
                                                            <svg className="w-4 h-4 mr-1" fill="none" stroke="currentColor" viewBox="0 0 24 24">
                                                                <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M9 12l2 2 4-4m6 2a9 9 0 11-18 0 9 9 0 0118 0z" />
                                                            </svg>
                                                            Unblock
                                                        </>
                                                    )}
                                                </button>
                                            )}
                                        </td>
                                    </tr>
                                ))}
                            </tbody>
                        </table>
                    </div>
                )}
            </div>
        </div>
    );
};

export default AllUsers;
