// client/src/services/api.js
import axios from 'axios';

// Force demo mode in production to avoid network errors
const isProduction = import.meta.env.PROD;
const isDemoMode = isProduction; // Always use demo mode in production

// Only use backend in development
const backendUrl = isProduction 
  ? null // No backend calls in production
  : (import.meta.env.VITE_API_BASE_URL || 'http://localhost:5000/api');

const api = axios.create({
  baseURL: backendUrl,
  headers: {
    'Content-Type': 'application/json',
  },
  // Add timeout and CORS support
  timeout: 10000,
  withCredentials: false,
});

// Add token to requests
api.interceptors.request.use(
  (config) => {
    const token = localStorage.getItem('token');
    if (token) {
      config.headers.Authorization = `Bearer ${token}`;
    }
    return config;
  },
  (error) => {
    return Promise.reject(error);
  }
);

// Handle token expiration and demo mode
api.interceptors.response.use(
  (response) => response,
  (error) => {
    // Better error handling
    if (error.code === 'ERR_NETWORK' || error.message?.includes('Network Error')) {
      console.warn('Network error detected, falling back to demo mode if needed');
      if (!isDemoMode) {
        // Show user-friendly error message
        if (typeof window !== 'undefined') {
          console.error('Network connection failed. Please check your internet connection.');
        }
      }
    }
    
    if (error.response?.status === 401) {
      localStorage.removeItem('token');
      if (typeof window !== 'undefined') {
        window.location.href = '/login';
      }
    }
    return Promise.reject(error);
  }
);

// Demo mode functions for frontend-only operation
export const demoMode = {
  // Simulate API responses
  mockUser: {
    id: 'demo-user-123',
    email: 'demo@microearn.com',
    name: 'Demo User',
    role: 'worker', // Default role for testing
    initialCoinsReceived: true,
    balance: 1000,
    createdAt: new Date().toISOString()
  },
  
  mockLogin: async (email, password) => {
    // Simulate API delay
    await new Promise(resolve => setTimeout(resolve, 1000));
    
    // Demo credentials for different roles
    const credentials = {
      'admin@microearn.com': { role: 'admin', name: 'Admin User', coins: 10000, initialCoinsReceived: true },
      'buyer@microearn.com': { role: 'buyer', name: 'Buyer User', coins: 5000, initialCoinsReceived: true },
      'worker@microearn.com': { role: 'worker', name: 'Worker User', coins: 1000, initialCoinsReceived: true },
      'demo@microearn.com': { role: 'worker', name: 'Demo User', coins: 1000, initialCoinsReceived: true }
    };
    
    if (credentials[email]) {
      const user = { ...demoMode.mockUser, ...credentials[email], email };
      return {
        success: true,
        user,
        token: 'demo-token-12345'
      };
    }
    
    return {
      success: false,
      error: 'Invalid demo credentials. Use admin@microearn.com/admin123, buyer@microearn.com/buyer123, worker@microearn.com/worker123, or demo@microearn.com/demo123'
    };
  },
  
  mockRegister: async (userData) => {
    await new Promise(resolve => setTimeout(resolve, 1000));
    return {
      success: true,
      user: {
        ...demoMode.mockUser,
        email: userData.email,
        name: userData.name || 'Demo User',
        role: 'worker', // Default role, can be changed later
        initialCoinsReceived: true
      },
      token: 'demo-token-12345'
    };
  },
};

// Export demo mode status for components to check
export { isDemoMode };

export default api;
