// client/src/services/firebase.config.js
// Import the functions you need from the SDKs you need
import { initializeApp } from "firebase/app";
import { getAuth, GoogleAuthProvider, FacebookAuthProvider, GithubAuthProvider } from 'firebase/auth';

// TODO: Add SDKs for Firebase products that you want to use
// https://firebase.google.com/docs/web/setup#available-libraries

// Your web app's Firebase configuration
const firebaseConfig = {
  apiKey: "AIzaSyDBMgyFx8V0ZkhKgO8NOGXJ6SUvx5iJc88",
  authDomain: "microearn-bf05f.firebaseapp.com",
  projectId: "microearn-bf05f",
  storageBucket: "microearn-bf05f.firebasestorage.app",
  messagingSenderId: "527392347630",
  appId: "1:527392347630:web:8342631acb320ff6e16aff"
};

// Check if Firebase is properly configured (now using hardcoded credentials)
const isFirebaseConfigured = true; // Firebase is configured with real credentials

let app;
let auth;
let googleProvider;
let facebookProvider;
let githubProvider;

try {
  app = initializeApp(firebaseConfig);
  auth = getAuth(app);
  googleProvider = new GoogleAuthProvider();
  facebookProvider = new FacebookAuthProvider();
  githubProvider = new GithubAuthProvider();
} catch (error) {
  console.warn('Firebase initialization error:', error);
  auth = null;
  googleProvider = null;
  facebookProvider = null;
  githubProvider = null;
}

export { auth, googleProvider, facebookProvider, githubProvider, isFirebaseConfigured };
export default app;
