// client/src/pages/worker/BrowseTasks.jsx
import { useEffect, useState } from 'react';
import { useSearchParams } from 'react-router-dom';
import { Link } from 'react-router-dom';
import api from '../../services/api';
import toast from 'react-hot-toast';
import { useAuth } from '../../context/AuthContext';

const BrowseTasks = () => {
  const { isDemoMode } = useAuth();

  // Demo tasks data - Define this first
  const demoTasks = [
    {
      _id: '1',
      title: 'Data Entry - Product Catalog',
      description: 'Enter 100 product details into our e-commerce system including names, prices, and descriptions.',
      category: 'Data Entry',
      budget: 50,
      deadline: '2024-02-15',
      difficulty: 'Easy',
      status: 'open',
      submissionsCount: 12,
      maxSubmissions: 20,
      requiredWorkers: 20,
      currentWorkers: 12,
      buyerName: 'TechStore Inc',
      createdAt: '2024-02-01T10:00:00Z'
    },
    {
      _id: '2',
      title: 'Content Writing - Blog Posts',
      description: 'Write 5 blog posts about digital marketing trends, 800 words each with SEO optimization.',
      category: 'Content Writing',
      budget: 120,
      deadline: '2024-02-20',
      difficulty: 'Medium',
      status: 'open',
      submissionsCount: 8,
      maxSubmissions: 15,
      requiredWorkers: 15,
      currentWorkers: 8,
      buyerName: 'Marketing Pro',
      createdAt: '2024-02-01T12:00:00Z'
    },
    {
      _id: '3',
      title: 'Image Tagging for AI Training',
      description: 'Tag 500 images with appropriate labels for machine learning dataset. Simple categorization task.',
      category: 'Data Annotation',
      budget: 80,
      deadline: '2024-02-18',
      difficulty: 'Easy',
      status: 'open',
      submissionsCount: 25,
      maxSubmissions: 30,
      requiredWorkers: 30,
      currentWorkers: 25,
      buyerName: 'AI Research Lab',
      createdAt: '2024-02-01T14:00:00Z'
    },
    {
      _id: '4',
      title: 'Social Media Content Creation',
      description: 'Create 20 social media posts for Instagram including captions and hashtags for a fashion brand.',
      category: 'Social Media',
      budget: 100,
      deadline: '2024-02-17',
      difficulty: 'Medium',
      status: 'open',
      submissionsCount: 15,
      maxSubmissions: 25,
      requiredWorkers: 25,
      currentWorkers: 15,
      buyerName: 'Fashion Forward',
      createdAt: '2024-02-01T16:00:00Z'
    },
    {
      _id: '5',
      title: 'Translation - English to Spanish',
      description: 'Translate 10 technical documents from English to Spanish. Must be fluent in both languages.',
      category: 'Translation',
      budget: 150,
      deadline: '2024-02-22',
      difficulty: 'Hard',
      status: 'open',
      submissionsCount: 5,
      maxSubmissions: 10,
      requiredWorkers: 10,
      currentWorkers: 5,
      buyerName: 'Global Tech Corp',
      createdAt: '2024-02-01T18:00:00Z'
    },
    {
      _id: '6',
      title: 'Product Reviews Writing',
      description: 'Write 30 authentic product reviews for various electronics. Must be detailed and helpful.',
      category: 'Content Writing',
      budget: 90,
      deadline: '2024-02-19',
      difficulty: 'Easy',
      status: 'open',
      submissionsCount: 18,
      maxSubmissions: 20,
      requiredWorkers: 20,
      currentWorkers: 18,
      buyerName: 'Review Hub',
      createdAt: '2024-02-01T20:00:00Z'
    },
    {
      _id: '7',
      title: 'Survey Participation - Research Study',
      description: 'Complete a 15-minute survey about consumer behavior and shopping preferences.',
      category: 'Surveys',
      budget: 25,
      deadline: '2024-02-16',
      difficulty: 'Easy',
      status: 'open',
      submissionsCount: 45,
      maxSubmissions: 100,
      requiredWorkers: 100,
      currentWorkers: 45,
      buyerName: 'Research Institute',
      createdAt: '2024-02-02T08:00:00Z'
    },
    {
      _id: '8',
      title: 'Video Transcription',
      description: 'Transcribe 5 hours of educational video content from English to text format.',
      category: 'Transcription',
      budget: 200,
      deadline: '2024-02-25',
      difficulty: 'Medium',
      status: 'open',
      submissionsCount: 7,
      maxSubmissions: 12,
      requiredWorkers: 12,
      currentWorkers: 7,
      buyerName: 'EduPlatform',
      createdAt: '2024-02-02T10:00:00Z'
    },
    {
      _id: '9',
      title: 'Website Testing - Bug Reports',
      description: 'Test a new e-commerce website and report 10 bugs with detailed descriptions and screenshots.',
      category: 'Testing',
      budget: 75,
      deadline: '2024-02-21',
      difficulty: 'Medium',
      status: 'open',
      submissionsCount: 20,
      maxSubmissions: 30,
      requiredWorkers: 30,
      currentWorkers: 20,
      buyerName: 'WebDev Agency',
      createdAt: '2024-02-02T12:00:00Z'
    },
    {
      _id: '10',
      title: 'Data Collection - Contact Information',
      description: 'Collect contact information for 50 local businesses in a specific industry category.',
      category: 'Data Collection',
      budget: 60,
      deadline: '2024-02-18',
      difficulty: 'Easy',
      status: 'open',
      submissionsCount: 30,
      maxSubmissions: 50,
      requiredWorkers: 50,
      currentWorkers: 30,
      buyerName: 'Business Directory',
      createdAt: '2024-02-02T14:00:00Z'
    },
    {
      _id: '11',
      title: 'Logo Design for Startups',
      description: 'Design 5 professional logos for different startup companies with modern aesthetics.',
      category: 'Design',
      budget: 180,
      deadline: '2024-02-24',
      difficulty: 'Hard',
      status: 'open',
      submissionsCount: 8,
      maxSubmissions: 15,
      requiredWorkers: 15,
      currentWorkers: 8,
      buyerName: 'Startup Hub',
      createdAt: '2024-02-02T16:00:00Z'
    },
    {
      _id: '12',
      title: 'Customer Support Responses',
      description: 'Write 50 professional customer support email responses for common inquiries.',
      category: 'Customer Service',
      budget: 85,
      deadline: '2024-02-20',
      difficulty: 'Medium',
      status: 'open',
      submissionsCount: 22,
      maxSubmissions: 35,
      requiredWorkers: 35,
      currentWorkers: 22,
      buyerName: 'Support Solutions',
      createdAt: '2024-02-02T18:00:00Z'
    },
    {
      _id: '13',
      title: 'PDF Document Formatting',
      description: 'Format and style 20 PDF documents according to specific brand guidelines and templates.',
      category: 'Document Processing',
      budget: 70,
      deadline: '2024-02-19',
      difficulty: 'Easy',
      status: 'open',
      submissionsCount: 16,
      maxSubmissions: 25,
      requiredWorkers: 25,
      currentWorkers: 16,
      buyerName: 'DocuPro Services',
      createdAt: '2024-02-02T20:00:00Z'
    },
    {
      _id: '14',
      title: 'Market Research Analysis',
      description: 'Analyze market trends for a specific industry and create a comprehensive 10-page report.',
      category: 'Research',
      budget: 250,
      deadline: '2024-02-28',
      difficulty: 'Hard',
      status: 'open',
      submissionsCount: 3,
      maxSubmissions: 8,
      requiredWorkers: 8,
      currentWorkers: 3,
      buyerName: 'Market Insights',
      createdAt: '2024-02-03T08:00:00Z'
    },
    {
      _id: '15',
      title: 'Social Media Management',
      description: 'Manage and grow Instagram account for a small business for 2 weeks with daily posts.',
      category: 'Social Media',
      budget: 160,
      deadline: '2024-02-23',
      difficulty: 'Medium',
      status: 'open',
      submissionsCount: 10,
      maxSubmissions: 20,
      requiredWorkers: 20,
      currentWorkers: 10,
      buyerName: 'Local Business',
      createdAt: '2024-02-03T10:00:00Z'
    },
    {
      _id: '16',
      title: 'Audio Editing - Podcast Production',
      description: 'Edit and enhance audio quality for 5 podcast episodes including intro/outro music.',
      category: 'Audio Editing',
      budget: 140,
      deadline: '2024-02-26',
      difficulty: 'Medium',
      status: 'open',
      submissionsCount: 6,
      maxSubmissions: 12,
      requiredWorkers: 12,
      currentWorkers: 6,
      buyerName: 'Podcast Network',
      createdAt: '2024-02-03T12:00:00Z'
    },
    {
      _id: '17',
      title: 'Excel Data Analysis',
      description: 'Analyze sales data in Excel spreadsheets and create charts and summary reports.',
      category: 'Data Analysis',
      budget: 110,
      deadline: '2024-02-22',
      difficulty: 'Medium',
      status: 'open',
      submissionsCount: 14,
      maxSubmissions: 25,
      requiredWorkers: 25,
      currentWorkers: 14,
      buyerName: 'Sales Analytics',
      createdAt: '2024-02-03T14:00:00Z'
    },
    {
      _id: '18',
      title: 'Virtual Assistant Tasks',
      description: 'Complete various administrative tasks including email management, scheduling, and research.',
      category: 'Virtual Assistant',
      budget: 95,
      deadline: '2024-02-21',
      difficulty: 'Easy',
      status: 'open',
      submissionsCount: 28,
      maxSubmissions: 40,
      requiredWorkers: 40,
      currentWorkers: 28,
      buyerName: 'Executive Support',
      createdAt: '2024-02-03T16:00:00Z'
    },
    {
      _id: '19',
      title: 'Mobile App Testing',
      description: 'Test a new mobile application on iOS and Android devices and provide detailed feedback.',
      category: 'Testing',
      budget: 130,
      deadline: '2024-02-25',
      difficulty: 'Medium',
      status: 'open',
      submissionsCount: 11,
      maxSubmissions: 20,
      requiredWorkers: 20,
      currentWorkers: 11,
      buyerName: 'App Developers',
      createdAt: '2024-02-03T18:00:00Z'
    },
    {
      _id: '20',
      title: 'Content Moderation',
      description: 'Review and moderate user-generated content for an online community platform.',
      category: 'Moderation',
      budget: 65,
      deadline: '2024-02-20',
      difficulty: 'Easy',
      status: 'open',
      submissionsCount: 35,
      maxSubmissions: 50,
      requiredWorkers: 50,
      currentWorkers: 35,
      buyerName: 'Community Platform',
      createdAt: '2024-02-03T20:00:00Z'
    },
    {
      _id: '21',
      title: 'YouTube Video Descriptions',
      description: 'Write SEO-optimized descriptions for 50 YouTube videos including keywords and tags.',
      category: 'Content Writing',
      budget: 75,
      deadline: '2024-02-24',
      difficulty: 'Easy',
      status: 'open',
      submissionsCount: 19,
      maxSubmissions: 30,
      requiredWorkers: 30,
      currentWorkers: 19,
      buyerName: 'YouTube Creators',
      createdAt: '2024-02-04T08:00:00Z'
    },
    {
      _id: '22',
      title: 'Database Entry - Customer Records',
      description: 'Enter 200 customer records into CRM system with accurate data validation.',
      category: 'Data Entry',
      budget: 85,
      deadline: '2024-02-23',
      difficulty: 'Easy',
      status: 'open',
      submissionsCount: 24,
      maxSubmissions: 35,
      requiredWorkers: 35,
      currentWorkers: 24,
      buyerName: 'CRM Solutions',
      createdAt: '2024-02-04T10:00:00Z'
    },
    {
      _id: '23',
      title: 'Instagram Story Design',
      description: 'Create 30 professional Instagram story templates for various business niches.',
      category: 'Design',
      budget: 120,
      deadline: '2024-02-26',
      difficulty: 'Medium',
      status: 'open',
      submissionsCount: 9,
      maxSubmissions: 18,
      requiredWorkers: 18,
      currentWorkers: 9,
      buyerName: 'Social Media Agency',
      createdAt: '2024-02-04T12:00:00Z'
    },
    {
      _id: '24',
      title: 'Product Photography',
      description: 'Take high-quality photos of 25 products for e-commerce website with white background.',
      category: 'Photography',
      budget: 200,
      deadline: '2024-02-28',
      difficulty: 'Medium',
      status: 'open',
      submissionsCount: 5,
      maxSubmissions: 10,
      requiredWorkers: 10,
      currentWorkers: 5,
      buyerName: 'E-commerce Store',
      createdAt: '2024-02-04T14:00:00Z'
    },
    {
      _id: '25',
      title: 'Email Marketing Campaign',
      description: 'Design and write 10 promotional emails for a product launch campaign.',
      category: 'Email Marketing',
      budget: 90,
      deadline: '2024-02-22',
      difficulty: 'Medium',
      status: 'open',
      submissionsCount: 13,
      maxSubmissions: 20,
      requiredWorkers: 20,
      currentWorkers: 13,
      buyerName: 'Marketing Agency',
      createdAt: '2024-02-04T16:00:00Z'
    },
    {
      _id: '26',
      title: 'App Store Optimization',
      description: 'Research and optimize app store listings for 5 mobile apps with keywords and descriptions.',
      category: 'App Marketing',
      budget: 150,
      deadline: '2024-02-25',
      difficulty: 'Hard',
      status: 'open',
      submissionsCount: 4,
      maxSubmissions: 8,
      requiredWorkers: 8,
      currentWorkers: 4,
      buyerName: 'App Publisher',
      createdAt: '2024-02-04T18:00:00Z'
    },
    {
      _id: '27',
      title: 'Voice Recording - Audiobook',
      description: 'Record clear audio narration for a 50-page children\'s storybook with character voices.',
      category: 'Voice Recording',
      budget: 180,
      deadline: '2024-02-27',
      difficulty: 'Hard',
      status: 'open',
      submissionsCount: 3,
      maxSubmissions: 6,
      requiredWorkers: 6,
      currentWorkers: 3,
      buyerName: 'Publishing House',
      createdAt: '2024-02-04T20:00:00Z'
    },
    {
      _id: '28',
      title: 'Website Content Writing',
      description: 'Write 15 SEO-optimized blog posts for a technology website with 1000+ words each.',
      category: 'Content Writing',
      budget: 160,
      deadline: '2024-02-26',
      difficulty: 'Medium',
      status: 'open',
      submissionsCount: 7,
      maxSubmissions: 14,
      requiredWorkers: 14,
      currentWorkers: 7,
      buyerName: 'Tech Blog',
      createdAt: '2024-02-05T08:00:00Z'
    },
    {
      _id: '29',
      title: 'Game Testing - Beta Testing',
      description: 'Test a new mobile game and provide detailed feedback on gameplay and bugs.',
      category: 'Testing',
      budget: 100,
      deadline: '2024-02-24',
      difficulty: 'Easy',
      status: 'open',
      submissionsCount: 32,
      maxSubmissions: 50,
      requiredWorkers: 50,
      currentWorkers: 32,
      buyerName: 'Game Studio',
      createdAt: '2024-02-05T10:00:00Z'
    },
    {
      _id: '30',
      title: 'Business Card Design',
      description: 'Design 20 professional business cards for different industries with modern styles.',
      category: 'Design',
      budget: 80,
      deadline: '2024-02-22',
      difficulty: 'Medium',
      status: 'open',
      submissionsCount: 17,
      maxSubmissions: 25,
      requiredWorkers: 25,
      currentWorkers: 17,
      buyerName: 'Design Agency',
      createdAt: '2024-02-05T12:00:00Z'
    },
    {
      _id: '31',
      title: 'Survey Data Analysis',
      description: 'Analyze survey responses from 1000 participants and create visual charts and insights.',
      category: 'Data Analysis',
      budget: 130,
      deadline: '2024-02-25',
      difficulty: 'Medium',
      status: 'open',
      submissionsCount: 8,
      maxSubmissions: 15,
      requiredWorkers: 15,
      currentWorkers: 8,
      buyerName: 'Research Company',
      createdAt: '2024-02-05T14:00:00Z'
    },
    {
      _id: '32',
      title: 'Facebook Ad Creation',
      description: 'Create 15 compelling Facebook advertisements with images and copy for various products.',
      category: 'Social Media',
      budget: 110,
      deadline: '2024-02-23',
      difficulty: 'Medium',
      status: 'open',
      submissionsCount: 21,
      maxSubmissions: 30,
      requiredWorkers: 30,
      currentWorkers: 21,
      buyerName: 'Digital Marketing',
      createdAt: '2024-02-05T16:00:00Z'
    },
    {
      _id: '33',
      title: 'Recipe Writing - Food Blog',
      description: 'Write 25 detailed recipes with cooking instructions and ingredient lists for food blog.',
      category: 'Content Writing',
      budget: 95,
      deadline: '2024-02-24',
      difficulty: 'Easy',
      status: 'open',
      submissionsCount: 26,
      maxSubmissions: 40,
      requiredWorkers: 40,
      currentWorkers: 26,
      buyerName: 'Food Blog',
      createdAt: '2024-02-05T18:00:00Z'
    },
    {
      _id: '34',
      title: 'Software Testing - QA Testing',
      description: 'Perform comprehensive testing on new software and document all bugs and issues.',
      category: 'Testing',
      budget: 140,
      deadline: '2024-02-27',
      difficulty: 'Hard',
      status: 'open',
      submissionsCount: 6,
      maxSubmissions: 12,
      requiredWorkers: 12,
      currentWorkers: 6,
      buyerName: 'Software Company',
      createdAt: '2024-02-05T20:00:00Z'
    },
    {
      _id: '35',
      title: 'LinkedIn Profile Optimization',
      description: 'Optimize 10 LinkedIn profiles for professionals with enhanced descriptions and keywords.',
      category: 'Social Media',
      budget: 70,
      deadline: '2024-02-22',
      difficulty: 'Easy',
      status: 'open',
      submissionsCount: 29,
      maxSubmissions: 45,
      requiredWorkers: 45,
      currentWorkers: 29,
      buyerName: 'Career Services',
      createdAt: '2024-02-06T08:00:00Z'
    },
    {
      _id: '36',
      title: 'Presentation Design',
      description: 'Create 10 professional PowerPoint presentations for business meetings with animations.',
      category: 'Design',
      budget: 120,
      deadline: '2024-02-25',
      difficulty: 'Medium',
      status: 'open',
      submissionsCount: 12,
      maxSubmissions: 20,
      requiredWorkers: 20,
      currentWorkers: 12,
      buyerName: 'Business Consulting',
      createdAt: '2024-02-06T10:00:00Z'
    },
    {
      _id: '37',
      title: 'Product Research - Market Analysis',
      description: 'Research and analyze 20 competing products and create comprehensive comparison reports.',
      category: 'Research',
      budget: 170,
      deadline: '2024-02-28',
      difficulty: 'Hard',
      status: 'open',
      submissionsCount: 4,
      maxSubmissions: 8,
      requiredWorkers: 8,
      currentWorkers: 4,
      buyerName: 'Product Development',
      createdAt: '2024-02-06T12:00:00Z'
    },
    {
      _id: '38',
      title: 'Video Editing - YouTube Content',
      description: 'Edit 10 YouTube videos with transitions, effects, and background music.',
      category: 'Video Editing',
      budget: 150,
      deadline: '2024-02-26',
      difficulty: 'Medium',
      status: 'open',
      submissionsCount: 9,
      maxSubmissions: 16,
      requiredWorkers: 16,
      currentWorkers: 9,
      buyerName: 'YouTube Channel',
      createdAt: '2024-02-06T14:00:00Z'
    },
    {
      _id: '39',
      title: 'Customer Feedback Collection',
      description: 'Collect and organize customer feedback from 100 reviews and create summary reports.',
      category: 'Data Collection',
      budget: 80,
      deadline: '2024-02-23',
      difficulty: 'Easy',
      status: 'open',
      submissionsCount: 31,
      maxSubmissions: 50,
      requiredWorkers: 50,
      currentWorkers: 31,
      buyerName: 'Customer Service',
      createdAt: '2024-02-06T16:00:00Z'
    },
    {
      _id: '40',
      title: 'Infographic Design',
      description: 'Design 15 informative infographics for various topics with data visualization.',
      category: 'Design',
      budget: 135,
      deadline: '2024-02-27',
      difficulty: 'Medium',
      status: 'open',
      submissionsCount: 11,
      maxSubmissions: 20,
      requiredWorkers: 20,
      currentWorkers: 11,
      buyerName: 'Content Marketing',
      createdAt: '2024-02-06T18:00:00Z'
    },
    {
      _id: '41',
      title: 'Translation - French to English',
      description: 'Translate 15 business documents from French to English with professional terminology.',
      category: 'Translation',
      budget: 140,
      deadline: '2024-02-26',
      difficulty: 'Hard',
      status: 'open',
      submissionsCount: 5,
      maxSubmissions: 10,
      requiredWorkers: 10,
      currentWorkers: 5,
      buyerName: 'International Business',
      createdAt: '2024-02-06T20:00:00Z'
    },
    {
      _id: '42',
      title: 'Twitter Content Creation',
      description: 'Create 50 engaging tweets with hashtags for a technology company\'s Twitter account.',
      category: 'Social Media',
      budget: 85,
      deadline: '2024-02-24',
      difficulty: 'Easy',
      status: 'open',
      submissionsCount: 27,
      maxSubmissions: 40,
      requiredWorkers: 40,
      currentWorkers: 27,
      buyerName: 'Tech Company',
      createdAt: '2024-02-07T08:00:00Z'
    },
    {
      _id: '43',
      title: 'Database Cleanup - Data Quality',
      description: 'Clean and standardize 5000 customer records in database with duplicate removal.',
      category: 'Data Processing',
      budget: 110,
      deadline: '2024-02-25',
      difficulty: 'Medium',
      status: 'open',
      submissionsCount: 8,
      maxSubmissions: 15,
      requiredWorkers: 15,
      currentWorkers: 8,
      buyerName: 'Data Management',
      createdAt: '2024-02-07T10:00:00Z'
    },
    {
      _id: '44',
      title: 'Brochure Design - Marketing Materials',
      description: 'Design 20 professional brochures for different products with compelling copy and visuals.',
      category: 'Design',
      budget: 125,
      deadline: '2024-02-27',
      difficulty: 'Medium',
      status: 'open',
      submissionsCount: 10,
      maxSubmissions: 18,
      requiredWorkers: 18,
      currentWorkers: 10,
      buyerName: 'Marketing Materials',
      createdAt: '2024-02-07T12:00:00Z'
    },
    {
      _id: '45',
      title: 'Online Research - Industry Trends',
      description: 'Research and compile information about latest industry trends for 10 different sectors.',
      category: 'Research',
      budget: 95,
      deadline: '2024-02-24',
      difficulty: 'Medium',
      status: 'open',
      submissionsCount: 15,
      maxSubmissions: 25,
      requiredWorkers: 25,
      currentWorkers: 15,
      buyerName: 'Market Research',
      createdAt: '2024-02-07T14:00:00Z'
    },
    {
      _id: '46',
      title: 'Podcast Show Notes',
      description: 'Write detailed show notes for 20 podcast episodes with timestamps and resources.',
      category: 'Content Writing',
      budget: 100,
      deadline: '2024-02-26',
      difficulty: 'Easy',
      status: 'open',
      submissionsCount: 18,
      maxSubmissions: 30,
      requiredWorkers: 30,
      currentWorkers: 18,
      buyerName: 'Podcast Production',
      createdAt: '2024-02-07T16:00:00Z'
    },
    {
      _id: '47',
      title: 'Website User Testing',
      description: 'Test website usability and provide detailed feedback on user experience and navigation.',
      category: 'Testing',
      budget: 90,
      deadline: '2024-02-25',
      difficulty: 'Easy',
      status: 'open',
      submissionsCount: 22,
      maxSubmissions: 35,
      requiredWorkers: 35,
      currentWorkers: 22,
      buyerName: 'UX Design Agency',
      createdAt: '2024-02-07T18:00:00Z'
    },
    {
      _id: '48',
      title: 'Business Plan Writing',
      description: 'Write comprehensive business plans for 3 startup ideas with financial projections.',
      category: 'Business Writing',
      budget: 200,
      deadline: '2024-02-28',
      difficulty: 'Hard',
      status: 'open',
      submissionsCount: 2,
      maxSubmissions: 5,
      requiredWorkers: 5,
      currentWorkers: 2,
      buyerName: 'Startup Accelerator',
      createdAt: '2024-02-07T20:00:00Z'
    },
    {
      _id: '49',
      title: 'Data Visualization - Dashboard Creation',
      description: 'Create interactive dashboards with charts and graphs for business analytics.',
      category: 'Data Visualization',
      budget: 160,
      deadline: '2024-02-27',
      difficulty: 'Hard',
      status: 'open',
      submissionsCount: 4,
      maxSubmissions: 8,
      requiredWorkers: 8,
      currentWorkers: 4,
      buyerName: 'Analytics Company',
      createdAt: '2024-02-08T08:00:00Z'
    },
    {
      _id: '50',
      title: 'Content Curation - Blog Posts',
      description: 'Curate and organize 50 relevant articles and resources for a weekly newsletter.',
      category: 'Content Curation',
      budget: 75,
      deadline: '2024-02-24',
      difficulty: 'Easy',
      status: 'open',
      submissionsCount: 33,
      maxSubmissions: 50,
      requiredWorkers: 50,
      currentWorkers: 33,
      buyerName: 'Newsletter Service',
      createdAt: '2024-02-08T10:00:00Z'
    }
  ];

  const [tasks, setTasks] = useState(demoTasks); // Initialize with demo tasks
  const [loading, setLoading] = useState(false); // Start with false since we have demo tasks
  const [pagination, setPagination] = useState({
    currentPage: 1,
    totalPages: 5, // 50 tasks / 10 per page = 5 pages
    totalTasks: 50,
    hasNextPage: true,
    hasPrevPage: false
  });

  // URL Params state
  const [searchParams, setSearchParams] = useSearchParams();
  const page = parseInt(searchParams.get('page')) || 1;
  const q = searchParams.get('q') || '';
  const category = searchParams.get('category') || '';
  const sort = searchParams.get('sort') || 'newest';

  useEffect(() => {
    console.log('BrowseTasks useEffect - isDemoMode:', isDemoMode);
    console.log('BrowseTasks useEffect - demoTasks length:', demoTasks.length);
    
    // Always use demo mode for now to ensure tasks show
    if (true || isDemoMode) {
      // In demo mode, use demo tasks with pagination
      console.log('Setting demo tasks:', demoTasks.length);
      
      // Calculate pagination for 10 items per page
      const itemsPerPage = 10;
      const startIndex = (page - 1) * itemsPerPage;
      const endIndex = startIndex + itemsPerPage;
      const paginatedTasks = demoTasks.slice(startIndex, endIndex);
      
      setTasks(paginatedTasks);
      setLoading(false);
      setPagination({
        currentPage: page,
        totalPages: Math.ceil(demoTasks.length / itemsPerPage), // 50 / 10 = 5 pages
        totalTasks: demoTasks.length,
        hasNextPage: page < Math.ceil(demoTasks.length / itemsPerPage),
        hasPrevPage: page > 1
      });
    } else {
      // In development, fetch from API
      console.log('Fetching from API');
      fetchTasks();
    }
  }, [page, q, category, sort, isDemoMode]);

  const fetchTasks = async () => {
    try {
      setLoading(true);
      const params = new URLSearchParams({
        page,
        limit: 9,
        q,
        category,
        sort,
      });
      // Filter out empty params
      if (!q) params.delete('q');
      if (!category) params.delete('category');

      const response = await api.get(`/tasks/available?${params.toString()}`);
      setTasks(response.data.tasks || []);
      setPagination(response.data.pagination || {});
    } catch (error) {
      toast.error('Failed to load tasks');
    } finally {
      setLoading(false);
    }
  };

  const handleSearch = (e) => {
    e.preventDefault();
    const formData = new FormData(e.target);
    const query = formData.get('search');

    setSearchParams(prev => {
      prev.set('q', query);
      prev.set('page', 1); // Reset to page 1 on search
      return prev;
    });
  };

  const handleFilterChange = (key, value) => {
    setSearchParams(prev => {
      if (value) prev.set(key, value);
      else prev.delete(key);
      prev.set('page', 1);
      return prev;
    });
  };

  const handlePageChange = (newPage) => {
    setSearchParams(prev => {
      prev.set('page', newPage);
      return prev;
    });
  };

  return (
    <div>
      <div className="flex flex-col md:flex-row justify-between items-center mb-6 gap-4">
        <h2 className="text-2xl font-bold">Find Tasks</h2>

        {/* Search Bar */}
        <form onSubmit={handleSearch} className="flex w-full md:w-auto">
          <input
            type="text"
            name="search"
            defaultValue={q}
            placeholder="Search tasks..."
            className="w-full md:w-64 px-4 py-2 border border-gray-300 rounded-l-lg focus:outline-none focus:ring-2 focus:ring-blue-500"
          />
          <button
            type="submit"
            className="px-4 py-2 bg-blue-600 text-white rounded-r-lg hover:bg-blue-700 transition"
          >
            Search
          </button>
        </form>
      </div>

      {/* Filters */}
      <div className="bg-white p-4 rounded-lg shadow mb-6 flex flex-wrap gap-4 items-center">
        <div className="flex items-center space-x-2">
          <span className="text-gray-600 font-medium">Filter by:</span>
          <select
            value={category}
            onChange={(e) => handleFilterChange('category', e.target.value)}
            className="border-gray-300 rounded-md shadow-sm focus:border-blue-500 focus:ring-blue-500"
          >
            <option value="">All Categories</option>
            <option value="social_media">Social Media</option>
            <option value="app_install">App Install</option>
            <option value="survey">Survey</option>
            <option value="content">Content Writing</option>
            <option value="other">Other</option>
          </select>
        </div>

        <div className="flex items-center space-x-2">
          <span className="text-gray-600 font-medium">Sort by:</span>
          <select
            value={sort}
            onChange={(e) => handleFilterChange('sort', e.target.value)}
            className="border-gray-300 rounded-md shadow-sm focus:border-blue-500 focus:ring-blue-500"
          >
            <option value="newest">Newest First</option>
            <option value="coins_high">Coins: High to Low</option>
            <option value="coins_low">Coins: Low to High</option>
          </select>
        </div>

        {(q || category || sort !== 'newest') && (
          <button
            onClick={() => setSearchParams({})}
            className="text-red-500 hover:text-red-700 text-sm font-medium"
          >
            Clear Filters
          </button>
        )}
      </div>

      {loading ? (
        <div className="flex items-center justify-center h-64">
          <div className="animate-spin rounded-full h-12 w-12 border-b-2 border-blue-600"></div>
        </div>
      ) : (
        <>
          <div className="mb-4 text-sm text-gray-600">
            Debug: Loading: {loading ? 'Yes' : 'No'}, Tasks: {tasks.length}, Demo Mode: {isDemoMode ? 'Yes' : 'No'}
          </div>
          {tasks.length === 0 ? (
            <div className="bg-white p-12 rounded-lg shadow text-center">
              <svg className="mx-auto h-12 w-12 text-gray-400" fill="none" viewBox="0 0 24 24" stroke="currentColor">
                <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M9.172 16.172a4 4 0 015.656 0M9 10h.01M15 10h.01M21 12a9 9 0 11-18 0 9 9 0 0118 0z" />
              </svg>
              <h3 className="mt-2 text-sm font-medium text-gray-900">No tasks found</h3>
              <p className="mt-1 text-sm text-gray-500">Try adjusting your search or filters.</p>
            </div>
          ) : (
            <>
              <div className="grid grid-cols-1 sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-4 xl:grid-cols-5 gap-6">
            {tasks.map((task) => (
              <div
                key={task._id}
                className="group relative bg-white rounded-2xl shadow-md hover:shadow-2xl border border-gray-100 overflow-hidden transition-all duration-500 flex flex-col h-full hover:-translate-y-2 hover:border-indigo-300"
              >
                {/* Gradient overlay for visual appeal */}
                <div className="absolute inset-0 bg-gradient-to-br from-indigo-500/5 via-transparent to-purple-500/5 opacity-0 group-hover:opacity-100 transition-opacity duration-300"></div>
                
                {/* Header Section */}
                <div className="relative p-6 pb-0">
                  <div className="flex items-start justify-between mb-4">
                    <div className="flex items-center space-x-2">
                      <div className={`w-3 h-3 rounded-full ${
                        task.status === 'open' ? 'bg-green-500' : 
                        task.status === 'in_progress' ? 'bg-yellow-500' : 
                        'bg-gray-400'
                      } animate-pulse`}></div>
                      <span className="text-xs font-semibold text-gray-500 uppercase tracking-wider">
                        {task.category || 'General'}
                      </span>
                    </div>
                    <div className="flex items-center space-x-1">
                      <svg className="w-4 h-4 text-gray-400" fill="none" stroke="currentColor" viewBox="0 0 24 24">
                        <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z" />
                      </svg>
                      <span className="text-xs text-gray-400">
                        {new Date(task.createdAt).toLocaleDateString(undefined, { month: 'short', day: 'numeric' })}
                      </span>
                    </div>
                  </div>

                  {/* Title */}
                  <h3 className="text-xl font-bold text-gray-900 mb-3 group-hover:text-indigo-600 transition-colors line-clamp-2 leading-tight" title={task.title}>
                    {task.title}
                  </h3>

                  {/* Description */}
                  <p className="text-gray-600 text-sm mb-4 line-clamp-2 leading-relaxed">
                    {task.description}
                  </p>
                </div>

                {/* Content Section */}
                <div className="flex-1 px-6 pb-0">
                  {/* Reward Section */}
                  <div className="bg-gradient-to-r from-green-50 to-emerald-50 rounded-xl p-3 mb-4 border border-green-100">
                    <div className="flex items-center justify-between">
                      <div className="flex items-center space-x-2">
                        <svg className="w-5 h-5 text-green-600" fill="none" stroke="currentColor" viewBox="0 0 24 24">
                          <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M12 8c-1.657 0-3 .895-3 2s1.343 2 3 2 3 .895 3 2-1.343 2-3 2m0-8c1.11 0 2.08.402 2.599 1M12 8V7m0 1v8m0 0v1m0-1c-1.11 0-2.08-.402-2.599-1M21 12a9 9 0 11-18 0 9 9 0 0118 0z" />
                        </svg>
                        <span className="text-xs font-medium text-green-700">Reward</span>
                      </div>
                      <div className="text-right">
                        <span className="text-2xl font-black text-green-600">{task.coinsPerWorker}</span>
                        <span className="text-xs font-medium text-green-700 ml-1">coins</span>
                      </div>
                    </div>
                  </div>

                  {/* Task Stats */}
                  <div className="grid grid-cols-2 gap-3 mb-4">
                    <div className="bg-blue-50 rounded-lg p-2 text-center">
                      <div className="text-xs font-medium text-blue-600 mb-1">Spots Left</div>
                      <div className={`text-lg font-bold ${
                        task.requiredWorkers - task.currentWorkers <= 5 ? 'text-red-600' : 'text-blue-600'
                      }`}>
                        {task.requiredWorkers - task.currentWorkers}
                      </div>
                      <div className="text-xs text-gray-500">of {task.requiredWorkers}</div>
                    </div>
                    <div className="bg-purple-50 rounded-lg p-2 text-center">
                      <div className="text-xs font-medium text-purple-600 mb-1">Difficulty</div>
                      <div className={`text-xs font-bold px-2 py-1 rounded-full inline-block ${
                        task.difficulty === 'Easy' ? 'bg-blue-100 text-blue-800' : 
                        task.difficulty === 'Medium' ? 'bg-orange-100 text-orange-800' : 
                        'bg-red-100 text-red-800'
                      }`}>
                        {task.difficulty}
                      </div>
                    </div>
                  </div>
                </div>

                {/* Footer Section */}
                <div className="px-6 pb-6 mt-auto">
                  <div className="flex items-center justify-between mb-3">
                    <span className={`inline-flex items-center px-3 py-1 rounded-full text-xs font-medium ${
                      task.status === 'open' ? 'bg-green-100 text-green-800' : 
                      task.status === 'in_progress' ? 'bg-yellow-100 text-yellow-800' : 
                      'bg-gray-100 text-gray-800'
                    }`}>
                      {task.status === 'open' ? '🟢 Open' : task.status === 'in_progress' ? '🟡 In Progress' : '⚪ Completed'}
                    </span>
                    <div className="flex items-center space-x-1">
                      <svg className="w-4 h-4 text-gray-400" fill="none" stroke="currentColor" viewBox="0 0 24 24">
                        <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M17 20h5v-2a3 3 0 00-5.356-1.857M17 20H7m10 0v-2c0-.656-.126-1.283-.356-1.857M7 20H2v-2a3 3 0 015.356-1.857M7 20v-2c0-.656.126-1.283.356-1.857m0 0a5.002 5.002 0 019.288 0M15 7a3 3 0 11-6 0 3 3 0 016 0zm6 3a2 2 0 11-4 0 2 2 0 014 0zM7 10a2 2 0 11-4 0 2 2 0 014 0z" />
                      </svg>
                      <span className="text-xs text-gray-500">{task.currentWorkers} workers</span>
                    </div>
                  </div>

                  <Link
                    to={`/worker/tasks/${task._id}`}
                    className="block w-full py-3 px-4 bg-gradient-to-r from-indigo-600 to-purple-600 text-white font-semibold rounded-xl hover:from-indigo-700 hover:to-purple-700 transition-all duration-300 text-center text-sm shadow-lg hover:shadow-xl transform hover:scale-[1.02] group-hover:shadow-indigo-500/25 cursor-pointer"
                  >
                    <span className="flex items-center justify-center space-x-2">
                      <span>View Task</span>
                      <svg className="w-4 h-4" fill="none" stroke="currentColor" viewBox="0 0 24 24">
                        <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M13 7l5 5m0 0l-5 5m5-5H6" />
                      </svg>
                    </span>
                  </Link>
                </div>
              </div>
            ))}
          </div>

          {/* Pagination */}
          {pagination.totalPages > 1 && (
            <div className="mt-8 flex justify-center items-center space-x-2">
              <button
                onClick={() => handlePageChange(page - 1)}
                disabled={!pagination.hasPrevPage}
                className="px-4 py-2 bg-white border border-gray-300 rounded-md shadow-sm text-sm font-medium text-gray-700 hover:bg-gray-50 disabled:opacity-50 disabled:cursor-not-allowed"
              >
                Previous
              </button>
              <span className="px-4 py-2 text-sm text-gray-700">
                Page {pagination.currentPage} of {pagination.totalPages}
              </span>
              <button
                onClick={() => handlePageChange(page + 1)}
                disabled={!pagination.hasNextPage}
                className="px-4 py-2 bg-white border border-gray-300 rounded-md shadow-sm text-sm font-medium text-gray-700 hover:bg-gray-50 disabled:opacity-50 disabled:cursor-not-allowed"
              >
                Next
              </button>
            </div>
          )}
            </>
          )}
        </>
      )}
    </div>
  );
};

export default BrowseTasks;
