// client/src/pages/worker/TaskDetail.jsx
import { useState, useEffect } from 'react';
import { useParams, useNavigate } from 'react-router-dom';
import api from '../../services/api';
import toast from 'react-hot-toast';
import { useAuth } from '../../context/AuthContext';

const TaskDetail = () => {
  const { id } = useParams();
  const navigate = useNavigate();
  const { isDemoMode } = useAuth();
  const [task, setTask] = useState(null);
  const [submission, setSubmission] = useState(null);
  const [submissionText, setSubmissionText] = useState('');
  const [submissionImage, setSubmissionImage] = useState('');
  const [loading, setLoading] = useState(true);
  const [submitting, setSubmitting] = useState(false);

  // Demo tasks data - matching BrowseTasks
  const demoTasks = [
    {
      _id: '1',
      title: 'Data Entry - Product Catalog',
      description: 'Enter 100 product details into our e-commerce system including names, prices, and descriptions. This task requires attention to detail and accuracy in data entry.',
      category: 'Data Entry',
      budget: 50,
      deadline: '2024-02-15',
      difficulty: 'Easy',
      status: 'open',
      submissionsCount: 12,
      maxSubmissions: 20,
      buyerName: 'TechStore Inc',
      createdAt: '2024-02-01T10:00:00Z',
      currentWorkers: 12,
      requiredWorkers: 20,
      coinsPerWorker: 50,
      submissionInstructions: 'Please enter product details in the provided spreadsheet format. Include product name, price, description, and category. Ensure accuracy and double-check all entries before submission.'
    },
    {
      _id: '2',
      title: 'Content Writing - Blog Posts',
      description: 'Write 5 blog posts about digital marketing trends, 800 words each with SEO optimization. Focus on current trends and provide actionable insights for readers.',
      category: 'Content Writing',
      budget: 120,
      deadline: '2024-02-20',
      difficulty: 'Medium',
      status: 'open',
      submissionsCount: 8,
      maxSubmissions: 15,
      buyerName: 'Marketing Pro',
      createdAt: '2024-02-01T12:00:00Z',
      currentWorkers: 8,
      requiredWorkers: 15,
      coinsPerWorker: 120,
      submissionInstructions: 'Write 5 blog posts, each 800 words minimum. Include SEO keywords, proper formatting, and actionable insights. Submit as a single document with clear separation between posts.'
    },
    {
      _id: '3',
      title: 'Image Tagging for AI Training',
      description: 'Tag and label 500 images for AI model training. Identify objects, scenes, and activities in each image with proper categorization.',
      category: 'AI Training',
      budget: 80,
      deadline: '2024-02-18',
      difficulty: 'Easy',
      status: 'open',
      submissionsCount: 30,
      maxSubmissions: 30,
      buyerName: 'AI Tech Corp',
      createdAt: '2024-02-01T14:00:00Z',
      currentWorkers: 30,
      requiredWorkers: 30,
      coinsPerWorker: 80,
      submissionInstructions: 'Use the provided tagging tool to label images. Follow the category guidelines and ensure accuracy in object identification.'
    },
    {
      _id: '4',
      title: 'Social Media Content Creation',
      description: 'Create 20 social media posts for Instagram and Facebook including captions, hashtags, and visual concepts.',
      category: 'Social Media',
      budget: 100,
      deadline: '2024-02-25',
      difficulty: 'Medium',
      status: 'open',
      submissionsCount: 15,
      maxSubmissions: 25,
      buyerName: 'Social Boost Agency',
      createdAt: '2024-02-01T16:00:00Z',
      currentWorkers: 15,
      requiredWorkers: 25,
      coinsPerWorker: 100,
      submissionInstructions: 'Create engaging content with relevant hashtags. Include both image concepts and text captions. Follow brand guidelines provided.'
    },
    {
      _id: '5',
      title: 'Translation - English to Spanish',
      description: 'Translate 10 technical documents from English to Spanish. Maintain technical accuracy and proper terminology.',
      category: 'Translation',
      budget: 150,
      deadline: '2024-02-22',
      difficulty: 'Hard',
      status: 'completed',
      submissionsCount: 10,
      maxSubmissions: 10,
      buyerName: 'Global Tech Solutions',
      createdAt: '2024-02-01T18:00:00Z',
      currentWorkers: 10,
      requiredWorkers: 10,
      coinsPerWorker: 150,
      submissionInstructions: 'Translate technical documents maintaining accuracy. Use proper technical terminology for Spanish-speaking markets.'
    },
    {
      _id: '6',
      title: 'Product Reviews Writing',
      description: 'Write 25 detailed product reviews for e-commerce platform. Each review should be 200-300 words with pros and cons.',
      category: 'Content Writing',
      budget: 75,
      deadline: '2024-02-19',
      difficulty: 'Easy',
      status: 'open',
      submissionsCount: 18,
      maxSubmissions: 22,
      buyerName: 'Review Hub',
      createdAt: '2024-02-02T08:00:00Z',
      currentWorkers: 18,
      requiredWorkers: 22,
      coinsPerWorker: 75,
      submissionInstructions: 'Write unbiased product reviews with pros and cons. Follow the review template and include star ratings.'
    },
    {
      _id: '7',
      title: 'Video Transcription',
      description: 'Transcribe 10 hours of video content from English to text. Include timestamps and speaker identification.',
      category: 'Transcription',
      budget: 90,
      deadline: '2024-02-24',
      difficulty: 'Medium',
      status: 'open',
      submissionsCount: 7,
      maxSubmissions: 12,
      buyerName: 'Media Productions',
      createdAt: '2024-02-02T10:00:00Z',
      currentWorkers: 7,
      requiredWorkers: 12,
      coinsPerWorker: 90,
      submissionInstructions: 'Transcribe videos accurately with timestamps. Identify different speakers and include non-verbal cues when relevant.'
    },
    {
      _id: '8',
      title: 'Survey Data Collection',
      description: 'Collect 100 survey responses for market research. Follow specific demographic requirements and ensure data quality.',
      category: 'Data Collection',
      budget: 60,
      deadline: '2024-02-21',
      difficulty: 'Easy',
      status: 'open',
      submissionsCount: 25,
      maxSubmissions: 30,
      buyerName: 'Research Insights',
      createdAt: '2024-02-02T12:00:00Z',
      currentWorkers: 25,
      requiredWorkers: 30,
      coinsPerWorker: 60,
      submissionInstructions: 'Collect survey responses following demographic guidelines. Ensure data quality and authenticity of responses.'
    },
    {
      _id: '9',
      title: 'App Testing & Bug Reports',
      description: 'Test mobile app functionality and report bugs. Provide detailed bug reports with screenshots and reproduction steps.',
      category: 'Testing',
      budget: 110,
      deadline: '2024-02-23',
      difficulty: 'Medium',
      status: 'open',
      submissionsCount: 9,
      maxSubmissions: 15,
      buyerName: 'App Developers Inc',
      createdAt: '2024-02-02T14:00:00Z',
      currentWorkers: 9,
      requiredWorkers: 15,
      coinsPerWorker: 110,
      submissionInstructions: 'Test all app features and document bugs with screenshots. Include reproduction steps and device information.'
    },
    {
      _id: '10',
      title: 'Email Campaign Setup',
      description: 'Set up email marketing campaigns using provided templates. Customize content and schedule delivery times.',
      category: 'Marketing',
      budget: 85,
      deadline: '2024-02-20',
      difficulty: 'Medium',
      status: 'open',
      submissionsCount: 14,
      maxSubmissions: 18,
      buyerName: 'Email Marketing Pro',
      createdAt: '2024-02-02T16:00:00Z',
      currentWorkers: 14,
      requiredWorkers: 18,
      coinsPerWorker: 85,
      submissionInstructions: 'Customize email templates and schedule campaigns. Follow brand guidelines and optimize for open rates.'
    }
  ];

  useEffect(() => {
    if (isDemoMode) {
      // In demo mode, find the task from demo data
      const demoTask = demoTasks.find(t => t._id === id);
      if (demoTask) {
        setTask(demoTask);
        setLoading(false);
      } else {
        toast.error('Task not found');
        navigate('/worker/tasks');
      }
    } else {
      fetchTask();
    }
  }, [id, isDemoMode]);

  const fetchTask = async () => {
    try {
      const response = await api.get(`/tasks/${id}`);
      setTask(response.data.task);
      setSubmission(response.data.userSubmission);
    } catch (error) {
      toast.error('Failed to load task');
      navigate('/worker/tasks');
    } finally {
      setLoading(false);
    }
  };

  const handleSubmit = async (e) => {
    e.preventDefault();
    
    // Basic validation for demo mode
    if (!isDemoMode && !submissionText.trim()) {
      toast.error('Please enter submission text');
      return;
    }

    setSubmitting(true);

    try {
      if (isDemoMode) {
        // In demo mode, simulate submission
        await new Promise(resolve => setTimeout(resolve, 1000)); // Simulate API call
        toast.success('Task submitted successfully! (Demo Mode)');
        navigate('/worker/my-submissions');
      } else {
        await api.post('/submissions', {
          taskId: id,
          submissionText,
          submissionImages: submissionImage ? [submissionImage] : [],
        });
        toast.success('Submission created successfully!');
        navigate('/worker/my-submissions');
      }
    } catch (error) {
      toast.error(error.response?.data?.message || 'Failed to submit');
    } finally {
      setSubmitting(false);
    }
  };

  if (loading) {
    return (
      <div className="flex items-center justify-center h-64">
        <div className="animate-spin rounded-full h-12 w-12 border-b-2 border-blue-600"></div>
      </div>
    );
  }

  if (!task) return null;

  return (
    <div className="max-w-7xl mx-auto space-y-8 animate-in fade-in slide-in-from-bottom-6 duration-700 p-6">
      {/* Breadcrumbs & Navigation */}
      <div className="flex items-center justify-between">
        <div className="flex items-center space-x-2 text-sm font-black uppercase tracking-widest text-gray-400">
          <button onClick={() => navigate('/worker/tasks')} className="hover:text-indigo-600 transition-colors">Find Tasks</button>
          <span>/</span>
          <span className="text-gray-900">Task Details</span>
        </div>
        <button 
          onClick={() => navigate('/worker/tasks')}
          className="px-4 py-2 bg-gray-100 text-gray-700 rounded-lg hover:bg-gray-200 transition-colors text-sm font-medium"
        >
          ← Back to Tasks
        </button>
      </div>

      {/* Task Header */}
      <div className="bg-white rounded-3xl shadow-xl border border-gray-100 overflow-hidden">
        <div className="bg-gradient-to-r from-indigo-600 to-purple-600 p-8 text-white">
          <div className="flex items-center justify-between mb-4">
            <div className="flex items-center space-x-3">
              <span className="px-4 py-2 bg-white/20 backdrop-blur-sm rounded-full text-sm font-bold uppercase tracking-wider">
                {task.category || 'General'}
              </span>
              <span className={`px-3 py-1 rounded-full text-xs font-medium ${
                task.status === 'open' ? 'bg-green-500 text-white' : 
                task.status === 'in_progress' ? 'bg-yellow-500 text-white' : 
                'bg-gray-500 text-white'
              }`}>
                {task.status === 'open' ? '🟢 Open' : task.status === 'in_progress' ? '🟡 In Progress' : '⚪ Completed'}
              </span>
            </div>
            <div className="text-right">
              <div className="text-3xl font-black">{task.coinsPerWorker}</div>
              <div className="text-sm opacity-90">Coins Reward</div>
            </div>
          </div>
          <h1 className="text-4xl lg:text-5xl font-black mb-4 leading-tight">
            {task.title}
          </h1>
          <div className="flex items-center space-x-6 text-sm opacity-90">
            <div className="flex items-center space-x-2">
              <svg className="w-4 h-4" fill="none" stroke="currentColor" viewBox="0 0 24 24">
                <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z" />
              </svg>
              <span>Deadline: {new Date(task.deadline).toLocaleDateString()}</span>
            </div>
            <div className="flex items-center space-x-2">
              <svg className="w-4 h-4" fill="none" stroke="currentColor" viewBox="0 0 24 24">
                <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M17 20h5v-2a3 3 0 00-5.356-1.857M17 20H7m10 0v-2c0-.656-.126-1.283-.356-1.857M7 20H2v-2a3 3 0 015.356-1.857M7 20v-2c0-.656.126-1.283.356-1.857m0 0a5.002 5.002 0 019.288 0M15 7a3 3 0 11-6 0 3 3 0 016 0zm6 3a2 2 0 11-4 0 2 2 0 014 0zM7 10a2 2 0 11-4 0 2 2 0 014 0z" />
              </svg>
              <span>{task.currentWorkers} / {task.requiredWorkers} Workers</span>
            </div>
            <div className="flex items-center space-x-2">
              <svg className="w-4 h-4" fill="none" stroke="currentColor" viewBox="0 0 24 24">
                <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M9 12l2 2 4-4m6 2a9 9 0 11-18 0 9 9 0 0118 0z" />
              </svg>
              <span>{task.submissionsCount} Submissions</span>
            </div>
          </div>
        </div>

        <div className="p-8">
          <div className="grid grid-cols-1 lg:grid-cols-3 gap-8">
            {/* Main Content */}
            <div className="lg:col-span-2 space-y-8">
              {/* Task Description */}
              <div>
                <h2 className="text-2xl font-bold text-gray-900 mb-4 flex items-center">
                  <svg className="w-6 h-6 mr-2 text-indigo-600" fill="none" stroke="currentColor" viewBox="0 0 24 24">
                    <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M13 16h-1v-4h-1m1-4h.01M21 12a9 9 0 11-18 0 9 9 0 0118 0z" />
                  </svg>
                  Task Description
                </h2>
                <div className="bg-gray-50 rounded-xl p-6 border border-gray-200">
                  <p className="text-gray-700 leading-relaxed text-lg">{task.description}</p>
                </div>
              </div>

              {/* Submission Instructions */}
              <div>
                <h2 className="text-2xl font-bold text-gray-900 mb-4 flex items-center">
                  <svg className="w-6 h-6 mr-2 text-indigo-600" fill="none" stroke="currentColor" viewBox="0 0 24 24">
                    <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M9 12h6m-6 4h6m2 5H7a2 2 0 01-2-2V5a2 2 0 012-2h5.586a1 1 0 01.707.293l5.414 5.414a1 1 0 01.293.707V19a2 2 0 01-2 2z" />
                  </svg>
                  Submission Instructions
                </h2>
                <div className="bg-blue-50 rounded-xl p-6 border border-blue-200">
                  <p className="text-blue-900 leading-relaxed">{task.submissionInstructions}</p>
                </div>
              </div>

              {/* Requirements */}
              <div>
                <h2 className="text-2xl font-bold text-gray-900 mb-4 flex items-center">
                  <svg className="w-6 h-6 mr-2 text-indigo-600" fill="none" stroke="currentColor" viewBox="0 0 24 24">
                    <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M9 5H7a2 2 0 00-2 2v12a2 2 0 002 2h10a2 2 0 002-2V7a2 2 0 00-2-2h-2M9 5a2 2 0 002 2h2a2 2 0 002-2M9 5a2 2 0 012-2h2a2 2 0 012 2" />
                  </svg>
                  Requirements
                </h2>
                <div className="grid grid-cols-1 md:grid-cols-2 gap-4">
                  <div className="bg-green-50 rounded-xl p-4 border border-green-200">
                    <div className="flex items-center space-x-2 mb-2">
                      <svg className="w-5 h-5 text-green-600" fill="none" stroke="currentColor" viewBox="0 0 24 24">
                        <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M5 13l4 4L19 7" />
                      </svg>
                      <span className="font-semibold text-green-900">Experience Level</span>
                    </div>
                    <p className="text-green-700">{task.difficulty} - Suitable for {task.difficulty === 'Easy' ? 'beginners' : task.difficulty === 'Medium' ? 'intermediate' : 'advanced'} workers</p>
                  </div>
                  <div className="bg-purple-50 rounded-xl p-4 border border-purple-200">
                    <div className="flex items-center space-x-2 mb-2">
                      <svg className="w-5 h-5 text-purple-600" fill="none" stroke="currentColor" viewBox="0 0 24 24">
                        <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z" />
                      </svg>
                      <span className="font-semibold text-purple-900">Time Commitment</span>
                    </div>
                    <p className="text-purple-700">Estimated {task.difficulty === 'Easy' ? '2-3 hours' : task.difficulty === 'Medium' ? '4-6 hours' : '8-12 hours'}</p>
                  </div>
                  <div className="bg-orange-50 rounded-xl p-4 border border-orange-200">
                    <div className="flex items-center space-x-2 mb-2">
                      <svg className="w-5 h-5 text-orange-600" fill="none" stroke="currentColor" viewBox="0 0 24 24">
                        <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M9 12h6m-6 4h6m2 5H7a2 2 0 01-2-2V5a2 2 0 012-2h5.586a1 1 0 01.707.293l5.414 5.414a1 1 0 01.293.707V19a2 2 0 01-2 2z" />
                      </svg>
                      <span className="font-semibold text-orange-900">Skills Required</span>
                    </div>
                    <p className="text-orange-700">Attention to detail, time management, basic computer skills</p>
                  </div>
                  <div className="bg-red-50 rounded-xl p-4 border border-red-200">
                    <div className="flex items-center space-x-2 mb-2">
                      <svg className="w-5 h-5 text-red-600" fill="none" stroke="currentColor" viewBox="0 0 24 24">
                        <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M12 8v4m0 4h.01M21 12a9 9 0 11-18 0 9 9 0 0118 0z" />
                      </svg>
                      <span className="font-semibold text-red-900">Quality Standards</span>
                    </div>
                    <p className="text-red-700">High accuracy, professional quality, on-time delivery</p>
                  </div>
                </div>
              </div>

              {/* Submission Form */}
              <div>
                <h2 className="text-2xl font-bold text-gray-900 mb-4 flex items-center">
                  <svg className="w-6 h-6 mr-2 text-indigo-600" fill="none" stroke="currentColor" viewBox="0 0 24 24">
                    <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M7 16a4 4 0 01-.88-7.903A5 5 0 1115.9 6L16 6a5 5 0 011 9.9M15 13l-3-3m0 0l-3 3m3-3v12" />
                  </svg>
                  Submit Your Work
                </h2>
                <form onSubmit={handleSubmit} className="bg-gray-50 rounded-xl p-6 border border-gray-200 space-y-6">
                  <div>
                    <label className="block text-sm font-medium text-gray-700 mb-2">
                      Submission Text
                    </label>
                    <textarea
                      value={submissionText}
                      onChange={(e) => setSubmissionText(e.target.value)}
                      rows={6}
                      className="w-full px-4 py-3 border border-gray-300 rounded-lg focus:ring-2 focus:ring-indigo-500 focus:border-indigo-500"
                      placeholder={isDemoMode ? "Describe your work and provide details about your submission... (Demo Mode - any text works)" : "Describe your work and provide details about your submission..."}
                      required={!isDemoMode}
                    />
                  </div>
                  
                  <div>
                    <label className="block text-sm font-medium text-gray-700 mb-2">
                      Supporting Image (Optional)
                    </label>
                    <input
                      type="url"
                      value={submissionImage}
                      onChange={(e) => setSubmissionImage(e.target.value)}
                      className="w-full px-4 py-3 border border-gray-300 rounded-lg focus:ring-2 focus:ring-indigo-500 focus:border-indigo-500"
                      placeholder="Enter image URL (optional)"
                    />
                  </div>

                  <button
                    type="submit"
                    disabled={submitting || (!isDemoMode && task.status !== 'open')}
                    className="w-full py-3 px-6 bg-gradient-to-r from-indigo-600 to-purple-600 text-white font-semibold rounded-xl hover:from-indigo-700 hover:to-purple-700 transition-all duration-300 disabled:opacity-50 disabled:cursor-not-allowed transform hover:scale-[1.02] active:scale-[0.98]"
                  >
                    {submitting ? (
                      <span className="flex items-center justify-center">
                        <svg className="animate-spin -ml-1 mr-3 h-5 w-5 text-white" xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24">
                          <circle className="opacity-25" cx="12" cy="12" r="10" stroke="currentColor" strokeWidth="4"></circle>
                          <path className="opacity-75" fill="currentColor" d="M4 12a8 8 0 018-8V0C5.373 0 0 5.373 0 12h4zm2 5.291A7.962 7.962 0 014 12H0c0 3.042 1.135 5.824 3 7.938l3-2.647z"></path>
                        </svg>
                        Submitting...
                      </span>
                    ) : (
                      <span className="flex items-center justify-center">
                        <svg className="w-5 h-5 mr-2" fill="none" stroke="currentColor" viewBox="0 0 24 24">
                          <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M7 16a4 4 0 01-.88-7.903A5 5 0 1115.9 6L16 6a5 5 0 011 9.9M15 13l-3-3m0 0l-3 3m3-3v12" />
                        </svg>
                        {isDemoMode ? 'Submit Task (Demo)' : 'Submit Task'}
                      </span>
                    )}
                  </button>
                </form>
              </div>
            </div>

            {/* Sidebar */}
            <div className="space-y-6">
              {/* Task Stats */}
              <div className="bg-white rounded-xl shadow-lg border border-gray-200 p-6">
                <h3 className="text-lg font-bold text-gray-900 mb-4">Task Statistics</h3>
                <div className="space-y-4">
                  <div className="flex justify-between items-center">
                    <span className="text-gray-600">Total Reward</span>
                    <span className="font-bold text-green-600 text-lg">{task.coinsPerWorker} coins</span>
                  </div>
                  <div className="flex justify-between items-center">
                    <span className="text-gray-600">Difficulty</span>
                    <span className={`px-3 py-1 rounded-full text-xs font-medium ${
                      task.difficulty === 'Easy' ? 'bg-blue-100 text-blue-800' : 
                      task.difficulty === 'Medium' ? 'bg-orange-100 text-orange-800' : 
                      'bg-red-100 text-red-800'
                    }`}>
                      {task.difficulty}
                    </span>
                  </div>
                  <div className="flex justify-between items-center">
                    <span className="text-gray-600">Status</span>
                    <span className={`px-3 py-1 rounded-full text-xs font-medium ${
                      task.status === 'open' ? 'bg-green-100 text-green-800' : 
                      task.status === 'in_progress' ? 'bg-yellow-100 text-yellow-800' : 
                      'bg-gray-100 text-gray-800'
                    }`}>
                      {task.status}
                    </span>
                  </div>
                  <div className="flex justify-between items-center">
                    <span className="text-gray-600">Posted Date</span>
                    <span className="text-sm">{new Date(task.createdAt).toLocaleDateString()}</span>
                  </div>
                  <div className="flex justify-between items-center">
                    <span className="text-gray-600">Deadline</span>
                    <span className="text-sm font-medium">{new Date(task.deadline).toLocaleDateString()}</span>
                  </div>
                </div>
              </div>

              {/* Progress Indicator */}
              <div className="bg-white rounded-xl shadow-lg border border-gray-200 p-6">
                <h3 className="text-lg font-bold text-gray-900 mb-4">Progress</h3>
                <div className="space-y-4">
                  <div>
                    <div className="flex justify-between items-center mb-2">
                      <span className="text-gray-600">Worker Slots</span>
                      <span className="text-sm font-medium">{task.currentWorkers} / {task.requiredWorkers}</span>
                    </div>
                    <div className="w-full bg-gray-200 rounded-full h-2">
                      <div 
                        className="bg-indigo-600 h-2 rounded-full transition-all duration-300"
                        style={{ width: `${(task.currentWorkers / task.requiredWorkers) * 100}%` }}
                      ></div>
                    </div>
                  </div>
                  <div>
                    <div className="flex justify-between items-center mb-2">
                      <span className="text-gray-600">Submissions</span>
                      <span className="text-sm font-medium">{task.submissionsCount}</span>
                    </div>
                    <div className="w-full bg-gray-200 rounded-full h-2">
                      <div 
                        className="bg-green-600 h-2 rounded-full transition-all duration-300"
                        style={{ width: `${Math.min((task.submissionsCount / task.requiredWorkers) * 100, 100)}%` }}
                      ></div>
                    </div>
                  </div>
                </div>
              </div>

              {/* Buyer Info */}
              <div className="bg-white rounded-xl shadow-lg border border-gray-200 p-6">
                <h3 className="text-lg font-bold text-gray-900 mb-4">Posted By</h3>
                <div className="flex items-center space-x-3">
                  <div className="w-12 h-12 bg-indigo-100 rounded-full flex items-center justify-center">
                    <svg className="w-6 h-6 text-indigo-600" fill="none" stroke="currentColor" viewBox="0 0 24 24">
                      <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M16 7a4 4 0 11-8 0 4 4 0 018 0zM12 14a7 7 0 00-7 7h14a7 7 0 00-7-7z" />
                    </svg>
                  </div>
                  <div>
                    <div className="font-semibold text-gray-900">{task.buyerName}</div>
                    <div className="text-sm text-gray-600">Verified Buyer</div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default TaskDetail;
