// client/src/context/AuthContext.jsx
import { createContext, useContext, useState, useEffect } from 'react';
import { auth, googleProvider, facebookProvider, githubProvider, isFirebaseConfigured } from '../services/firebase.config';
import {
  signInWithEmailAndPassword,
  createUserWithEmailAndPassword,
  signInWithPopup,
  signOut as firebaseSignOut,
  onAuthStateChanged,
  fetchSignInMethodsForEmail,
  updateProfile,
} from 'firebase/auth';
import api, { demoMode, isDemoMode } from '../services/api';

// Check if we're in production
const isProduction = import.meta.env.PROD;

const AuthContext = createContext();

export const useAuth = () => {
  const context = useContext(AuthContext);
  if (!context) {
    throw new Error('useAuth must be used within an AuthProvider');
  }
  return context;
};

export { isFirebaseConfigured } from '../services/firebase.config';

export const AuthProvider = ({ children }) => {
  const [user, setUser] = useState(null);
  const [loading, setLoading] = useState(true);
  const [token, setToken] = useState(localStorage.getItem('token'));

  useEffect(() => {
    // Check if we're in demo mode
    if (isDemoMode) {
      // In demo mode, check for existing demo token
      const existingToken = localStorage.getItem('token');
      if (existingToken) {
        setUser(demoMode.mockUser);
        setToken(existingToken);
      }
      setLoading(false);
      return;
    }

    if (!auth || !isFirebaseConfigured) {
      // If Firebase is not configured, just check for existing token
      const existingToken = localStorage.getItem('token');
      if (existingToken) {
        // Try to verify token with backend
        api.get('/auth/me')
          .then((response) => {
            setUser(response.data.user);
            setToken(existingToken);
          })
          .catch(() => {
            localStorage.removeItem('token');
            setUser(null);
            setToken(null);
          })
          .finally(() => {
            setLoading(false);
          });
      } else {
        setLoading(false);
      }
      return;
    }

    const unsubscribe = onAuthStateChanged(auth, async (firebaseUser) => {
      // Skip Firebase auth in demo mode
      if (isDemoMode) {
        return;
      }
      
      if (firebaseUser) {
        try {
          const idToken = await firebaseUser.getIdToken();
          const response = await api.post('/auth/verify-firebase', {
            idToken,
          });
          const { token: jwtToken, user: userData } = response.data;
          setToken(jwtToken);
          localStorage.setItem('token', jwtToken);
          setUser(userData);
        } catch (error) {
          console.error('Error verifying Firebase token:', error);
          setUser(null);
          setToken(null);
          localStorage.removeItem('token');
        }
      } else {
        setUser(null);
        setToken(null);
        localStorage.removeItem('token');
      }
      setLoading(false);
    });

    return unsubscribe;
  }, []);

  const login = async (email, password) => {
    // Check if we're in demo mode (always true in production)
    if (isDemoMode) {
      const result = await demoMode.mockLogin(email, password);
      if (result.success) {
        setToken(result.token);
        localStorage.setItem('token', result.token);
        setUser(result.user);
      }
      return result;
    }
    
    // Only try Firebase in development
    if (!isProduction && (!auth || !isFirebaseConfigured)) {
      return { success: false, error: 'Firebase authentication is not available. Please try demo credentials: demo@microearn.com / demo123' };
    }
    
    if (isProduction) {
      // In production, always use demo mode
      const result = await demoMode.mockLogin(email, password);
      if (result.success) {
        setToken(result.token);
        localStorage.setItem('token', result.token);
        setUser(result.user);
      }
      return result;
    }
    
    try {
      const userCredential = await signInWithEmailAndPassword(
        auth,
        email,
        password
      );
      const idToken = await userCredential.user.getIdToken();
      
      // Try to verify with backend, but fall back to demo mode if it fails
      try {
        const response = await api.post('/auth/verify-firebase', { idToken });
        const { token: jwtToken, user: userData } = response.data;
        setToken(jwtToken);
        localStorage.setItem('token', jwtToken);
        setUser(userData);
        return { success: true, user: userData };
      } catch (backendError) {
        console.warn('Backend verification failed, using Firebase user directly:', backendError);
        // Fall back to using Firebase user directly
        const userData = {
          id: userCredential.user.uid,
          email: userCredential.user.email,
          name: userCredential.user.displayName || 'User',
          role: null,
          initialCoinsReceived: false,
          coins: 0,
          createdAt: userCredential.user.metadata.creationTime
        };
        setUser(userData);
        return { success: true, user: userData };
      }
    } catch (error) {
      console.error('Login error:', error);
      
      // Better error handling
      let errorMessage = 'Login failed. Please try again.';
      if (error.code === 'auth/user-not-found') {
        errorMessage = 'User not found. Please check your email address.';
      } else if (error.code === 'auth/wrong-password') {
        errorMessage = 'Incorrect password. Please try again.';
      } else if (error.code === 'auth/too-many-requests') {
        errorMessage = 'Too many login attempts. Please try again later.';
      } else if (error.code === 'ERR_NETWORK' || error.message?.includes('Network Error')) {
        errorMessage = 'Network connection failed. Please check your internet connection and try again.';
      } else if (error.response?.status === 500) {
        errorMessage = 'Server error. Please try again later.';
      }
      
      return { success: false, error: errorMessage };
    }
  };

  const register = async (email, password, displayName) => {
    // Check if we're in demo mode (always true in production)
    if (isDemoMode) {
      const result = await demoMode.mockRegister({ email, name: displayName });
      if (result.success) {
        setToken(result.token);
        localStorage.setItem('token', result.token);
        setUser(result.user);
      }
      return result;
    }
    
    // Only try Firebase in development
    if (!isProduction && (!auth || !isFirebaseConfigured)) {
      return { success: false, error: 'Firebase authentication is not available. Please try demo credentials: demo@microearn.com / demo123' };
    }
    
    if (isProduction) {
      // In production, always use demo mode
      const result = await demoMode.mockRegister({ email, name: displayName });
      if (result.success) {
        setToken(result.token);
        localStorage.setItem('token', result.token);
        setUser(result.user);
      }
      return result;
    }
    
    try {
      const userCredential = await createUserWithEmailAndPassword(
        auth,
        email,
        password
      );
      if (displayName) {
        await updateProfile(userCredential.user, { displayName });
      }
      const idToken = await userCredential.user.getIdToken();
      
      // Try to verify with backend, but fall back to demo mode if it fails
      try {
        const response = await api.post('/auth/verify-firebase', { idToken });
        const { token: jwtToken, user: userData } = response.data;
        setToken(jwtToken);
        localStorage.setItem('token', jwtToken);
        setUser(userData);
        return { success: true, user: userData };
      } catch (backendError) {
        console.warn('Backend verification failed, using Firebase user directly:', backendError);
        // Fall back to using Firebase user directly
        const userData = {
          id: userCredential.user.uid,
          email: userCredential.user.email,
          name: userCredential.user.displayName || displayName || 'User',
          role: null,
          initialCoinsReceived: false,
          coins: 0,
          createdAt: userCredential.user.metadata.creationTime
        };
        setUser(userData);
        return { success: true, user: userData };
      }
    } catch (error) {
      console.error('Register error:', error);
      
      // Better error handling
      let errorMessage = 'Registration failed. Please try again.';
      if (error.code === 'auth/email-already-in-use') {
        errorMessage = 'Email already exists. Please try logging in or use a different email.';
      } else if (error.code === 'auth/weak-password') {
        errorMessage = 'Password is too weak. Please choose a stronger password.';
      } else if (error.code === 'auth/invalid-email') {
        errorMessage = 'Invalid email address. Please check and try again.';
      } else if (error.code === 'ERR_NETWORK' || error.message?.includes('Network Error')) {
        errorMessage = 'Network connection failed. Please check your internet connection and try again.';
      } else if (error.response?.status === 500) {
        errorMessage = 'Server error. Please try again later.';
      }
      
      return { success: false, error: errorMessage };
    }
  };

  const loginWithSocial = async (providerName) => {
    // Check if we're in demo mode
    if (isDemoMode) {
      return await demoMode.mockLogin('demo@microearn.com', 'demo123');
    }
    
    if (!auth || !isFirebaseConfigured) {
      return { success: false, error: 'Firebase authentication is not available' };
    }

    let provider;
    if (providerName === 'google') provider = googleProvider;
    else if (providerName === 'facebook') provider = facebookProvider;
    else if (providerName === 'github') provider = githubProvider;

    if (!provider) {
      return { success: false, error: `Provider ${providerName} is not initialized.` };
    }

    try {
      const result = await signInWithPopup(auth, provider);
      const idToken = await result.user.getIdToken();
      const response = await api.post('/auth/verify-firebase', { idToken });
      const { token: jwtToken, user: userData } = response.data;
      setToken(jwtToken);
      localStorage.setItem('token', jwtToken);
      setUser(userData);
      return { success: true, user: userData };
    } catch (error) {
      console.error(`${providerName} login error:`, error);
      return { success: false, error: error.message };
    }
  };

  const selectRole = async (role) => {
    // Check if we're in demo mode
    if (isDemoMode) {
      const updatedUser = { ...demoMode.mockUser, role };
      setUser(updatedUser);
      return { success: true, user: updatedUser };
    }
    
    try {
      const response = await api.post('/auth/select-role', { role });
      const { token: jwtToken, user: userData } = response.data;
      setToken(jwtToken);
      localStorage.setItem('token', jwtToken);
      setUser(userData);
      return { success: true, user: userData };
    } catch (error) {
      console.error('Role selection error:', error);
      return { success: false, error: error.message };
    }
  };

  const logout = async () => {
    try {
      if (auth && isFirebaseConfigured) {
        await firebaseSignOut(auth);
      }
      setUser(null);
      setToken(null);
      localStorage.removeItem('token');
    } catch (error) {
      console.error('Logout error:', error);
      // Still clear local state even if Firebase logout fails
      setUser(null);
      setToken(null);
      localStorage.removeItem('token');
    }
  };

  const value = {
    user,
    token,
    loading,
    login,
    register,
    loginWithGoogle: () => loginWithSocial('google'),
    loginWithFacebook: () => loginWithSocial('facebook'),
    loginWithGithub: () => loginWithSocial('github'),
    selectRole,
    logout,
  };

  return <AuthContext.Provider value={value}>{children}</AuthContext.Provider>;
};
